// @scalar/themes@0.5.4/dist/index.js
(function(){"use strict";try{if(typeof document<"u"){var e=document.createElement("style");e.appendChild(document.createTextNode('@import"https://fonts.googleapis.com/css2?family=Inter:wght@100..900";@import"https://fonts.googleapis.com/css?family=JetBrains%20Mono";:root{--default-theme-border-width: 1px;--default-theme-radius: 3px;--default-theme-radius-lg: 6px;--default-theme-radius-xl: 8px;--default-theme-header-height: 50px;--default-theme-sidebar-width: 250px;--default-theme-toc-width: 250px;--default-theme-font: "Inter", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;--default-theme-font-code: "JetBrains Mono", ui-monospace, Menlo, Monaco, "Cascadia Mono", "Segoe UI Mono", "Roboto Mono", "Oxygen Mono", "Ubuntu Monospace", "Source Code Pro", "Fira Mono", "Droid Sans Mono", "Courier New", monospace;--default-theme-heading-1: 40px;--default-theme-page-description: 24px;--default-theme-heading-2: 24px;--default-theme-heading-3: 20px;--default-theme-heading-4: 16px;--default-theme-heading-5: 16px;--default-theme-heading-6: 16px;--default-theme-paragraph: 16px;--default-theme-small: 14px;--default-theme-mini: 13px;--default-theme-micro: 12px;--default-theme-bold: 600;--default-theme-semibold: 500;--default-theme-regular: 400;--default-theme-font-size-1: 24px;--default-theme-font-size-2: 16px;--default-theme-font-size-3: 14px;--default-theme-font-size-4: 13px;--default-theme-font-size-5: 12px;--default-theme-line-height-1: 32px;--default-theme-line-height-2: 24px;--default-theme-line-height-3: 20px;--default-theme-line-height-4: 18px;--default-theme-line-height-5: 16px;--default-theme-font-medium: 500;--default-theme-font-bold: 700}@media (max-width: 460px){:root{--default-theme-font-size-1: 22px;--default-theme-font-size-2: 14px;--default-theme-font-size-3: 12px}}@media (max-width: 720px){:root{--default-theme-heading-1: 24px;--default-theme-page-description: 20px}}.cm-scroller,.custom-scroll{overflow-y:scroll;scrollbar-color:transparent transparent;scrollbar-width:thin;-webkit-overflow-scrolling:touch}.custom-scroll-self-contain-overflow{overscroll-behavior:contain}.custom-scroll-extra-padding-right{padding-right:12px!important}@supports (-moz-appearance: none){.cm-scroller,.custom-scroll{padding-right:12px}.custom-scroll-extra-padding-right{padding-right:24px!important}}.cm-scroller:hover,.custom-scroll:hover{scrollbar-color:var( --theme-scrollbar-color, var(--default-theme-scrollbar-color) ) transparent}.cm-scroller:hover::-webkit-scrollbar-thumb,.custom-scroll:hover::-webkit-scrollbar-thumb{background:var( --theme-scrollbar-color, var(--default-theme-scrollbar-color) );background-clip:content-box;border:3px solid transparent}.cm-scroller::-webkit-scrollbar-thumb:active,.custom-scroll::-webkit-scrollbar-thumb:active{background:var( --theme-scrollbar-color-active, var(--default-theme-scrollbar-color-active) );background-clip:content-box;border:3px solid transparent}.cm-scroller::-webkit-scrollbar-corner,.custom-scroll::-webkit-scrollbar-corner{background:transparent}.cm-scroller::-webkit-scrollbar,.custom-scroll::-webkit-scrollbar{height:12px;width:12px}.cm-scroller::-webkit-scrollbar-track,.custom-scroll::-webkit-scrollbar-track{background:transparent}.cm-scroller::-webkit-scrollbar-thumb,.custom-scroll::-webkit-scrollbar-thumb{border-radius:20px;background:transparent;background-clip:content-box;border:3px solid transparent}@media (pointer: coarse){.cm-scroller,.custom-scroll{padding-right:12px}.custom-scroll-extra-padding-right{padding-right:24px!important}}')),document.head.appendChild(e)}}catch(t){console.error("vite-plugin-css-injected-by-js",t)}})();
const br = `.light-mode{color-scheme:light;--default-theme-color-1: #24293c;--default-theme-color-2: #4a5166;--default-theme-color-3: #888c9a;--default-theme-color-accent: #0099ff;--default-theme-background-1: #fff;--default-theme-background-2: #f6f6f6;--default-theme-background-3: #e7e7e7;--default-theme-background-accent: #5369d20f;--default-theme-border-color: rgba(0, 0, 0, .1);--default-theme-scrollbar-color: rgba(0, 0, 0, .18);--default-theme-scrollbar-color-active: rgba(0, 0, 0, .36);--default-theme-lifted-brightness: 1;--default-theme-backdrop-brightness: 1;--default-theme-shadow-1: 0 1px 3px 0 rgba(0, 0, 0, .11);--default-theme-shadow-2: rgba(0, 0, 0, .08) 0px 13px 20px 0px, rgba(0, 0, 0, .08) 0px 3px 8px 0px, var(--default-theme-border-color) 0px 0 0 1px;--default-theme-button-1: rgb(49 53 56);--default-theme-button-1-color: #fff;--default-theme-button-1-hover: rgb(28 31 33);--default-theme-color-green: #069061;--default-theme-color-red: #ef0006;--default-theme-color-yellow: #edbe20;--default-theme-color-blue: #0082d0;--default-theme-color-orange: #fb892c;--default-theme-color-purple: #5203d1}.dark-mode{color-scheme:dark;--default-theme-color-1: rgba(255, 255, 255, .9);--default-theme-color-2: rgba(255, 255, 255, .62);--default-theme-color-3: rgba(255, 255, 255, .44);--default-theme-color-disabled: rgba(255, 255, 255, .34);--default-theme-color-ghost: rgba(255, 255, 255, .26);--default-theme-color-accent: #8ab4f8;--default-theme-background-1: #1a1a1a;--default-theme-background-2: #252525;--default-theme-background-3: #323232;--default-theme-background-4: rgba(255, 255, 255, .06);--default-theme-background-accent: #8ab4f81f;--default-theme-border-color: rgba(255, 255, 255, .1);--default-theme-scrollbar-color: rgba(255, 255, 255, .24);--default-theme-scrollbar-color-active: rgba(255, 255, 255, .48);--default-theme-lifted-brightness: 1.45;--default-theme-backdrop-brightness: .5;--default-theme-shadow-1: 0 1px 3px 0 rgb(0, 0, 0, .1);--default-theme-shadow-2: rgba(15, 15, 15, .2) 0px 3px 6px, rgba(15, 15, 15, .4) 0px 9px 24px, 0 0 0 1px rgba(255, 255, 255, .1);--default-theme-button-1: #f6f6f6;--default-theme-button-1-color: #000;--default-theme-button-1-hover: #e7e7e7;--default-theme-color-green: #00b648;--default-theme-color-red: #dc1b19;--default-theme-color-yellow: #ffc90d;--default-theme-color-blue: #4eb3ec;--default-theme-color-orange: #ff8d4d;--default-theme-color-purple: #b191f9;--default-theme-code-languages-background-supersede: transparent}.light-mode .t-doc__sidebar{--default-sidebar-background-1: var(--default-theme-background-2);--default-sidebar-item-hover-color: currentColor;--default-sidebar-item-hover-background: var(--default-theme-background-3);--default-sidebar-item-active-background: var( --default-theme-background-accent );--default-sidebar-border-color: transparent;--default-sidebar-color-1: var(--default-theme-color-1);--default-sidebar-color-2: var(--default-theme-color-2);--default-sidebar-color-active: var(--default-theme-color-accent);--default-sidebar-search-background: var(--default-theme-background-3);--default-sidebar-search-border-color: var(--default-theme-background-3);--default-sidebar-search--color: var(--default-theme-color-3)}.dark-mode .sidebar{--default-sidebar-background-1: var(--default-theme-background-2);--default-sidebar-item-hover-color: currentColor;--default-sidebar-item-hover-background: var(--default-theme-background-3);--default-sidebar-item-active-background: var( --default-theme-background-accent );--default-sidebar-border-color: var(--default-theme-background-2);--default-sidebar-color-1: var(--default-theme-color-1);--default-sidebar-color-2: var(--default-theme-color-2);--default-sidebar-color-active: var(--default-theme-color-accent);--default-sidebar-search-background: var(--default-theme-background-3);--default-sidebar-search-border-color: var(--default-theme-background-3);--default-sidebar-search--color: var(--default-theme-color-3)}.light-mode .t-doc__header{--default-header-background-1: #000;--default-header-border-color: #000;--default-header-color-1: #fff;--default-header-color-2: rgba(255, 255, 255, .66);--default-header-background-toggle: var(--default-theme-color-3);--default-header-call-to-action-color: var(--default-theme-color-accent)}.dark-mode .t-doc__header{--default-header-background-1: var(--default-theme-background-3);--default-header-border-color: var(--header-background-1);--default-header-color-1: var(--default-theme-color-1);--default-header-color-2: var(--default-theme-color-2);--default-header-background-toggle: var(--default-theme-color-3);--default-header-call-to-action-color: var(--default-theme-color-accent)}
`, mr = `.light-mode{--default-theme-color-1: rgb(9, 9, 11);--default-theme-color-2: rgb(113, 113, 122);--default-theme-color-3: rgba(25, 25, 28, .5);--default-theme-color-accent: var(--default-theme-color-1);--default-theme-background-1: #fff;--default-theme-background-2: #f4f4f5;--default-theme-background-3: #e3e3e6;--default-theme-background-accent: #8ab4f81f;--default-theme-border-color: rgb(228, 228, 231);--default-theme-code-language-color-supersede: var(--default-theme-color-1)}.dark-mode{--default-theme-color-1: #fafafa;--default-theme-color-2: rgb(161, 161, 170);--default-theme-color-3: rgba(255, 255, 255, .533);--default-theme-color-accent: var(--default-theme-color-1);--default-theme-background-1: #000e23;--default-theme-background-2: #01132e;--default-theme-background-3: #03193b;--default-theme-background-accent: #8ab4f81f;--default-theme-border-color: rgba(255, 255, 255, .12);--default-theme-code-language-color-supersede: var(--default-theme-color-1)}.light-mode .t-doc__sidebar,.dark-mode .t-doc__sidebar{--sidebar-background-1: transparent;--sidebar-item-hover-color: currentColor;--sidebar-item-hover-background: var(--default-theme-background-2);--sidebar-item-active-background: var(--default-theme-background-3);--sidebar-border-color: var(--default-theme-border-color);--sidebar-color-1: var(--default-theme-color-1);--sidebar-color-2: var(--default-theme-color-2);--sidebar-color-active: var(--default-theme-color-accent);--sidebar-search-background: rgba(255, 255, 255, .1);--sidebar-search-border-color: var(--default-theme-border-color);--sidebar-search--color: var(--default-theme-color-3);z-index:1}.light-mode .t-doc__sidebar{--sidebar-search-background: white}.light-mode{--default-theme-button-1: rgb(49 53 56);--default-theme-button-1-color: #fff;--default-theme-button-1-hover: rgb(28 31 33);--default-theme-color-green: #069061;--default-theme-color-red: #ef0006;--default-theme-color-yellow: #edbe20;--default-theme-color-blue: #0082d0;--default-theme-color-orange: #fb892c;--default-theme-color-purple: #5203d1;--default-theme-scrollbar-color: rgba(0, 0, 0, .18);--default-theme-scrollbar-color-active: rgba(0, 0, 0, .36)}.dark-mode{--default-theme-button-1: #f6f6f6;--default-theme-button-1-color: #000;--default-theme-button-1-hover: #e7e7e7;--default-theme-color-green: rgba(69, 255, 165, .823);--default-theme-color-red: #ff8589;--default-theme-color-yellow: #ffcc4d;--default-theme-color-blue: #6bc1fe;--default-theme-color-orange: #f98943;--default-theme-color-purple: #b191f9;--default-theme-scrollbar-color: rgba(255, 255, 255, .24);--default-theme-scrollbar-color-active: rgba(255, 255, 255, .48)}.show-api-client-button:before{background:white!important}.show-api-client-button span,.show-api-client-button svg{color:#000!important}.download-cta,.references-rendered .markdown a{text-decoration:underline!important}@keyframes headerbackground{0%{background:transparent;-webkit-backdrop-filter:none;backdrop-filter:none}to{background:var(--header-background-1);-webkit-backdrop-filter:blur(12px);backdrop-filter:blur(12px)}}.light-mode .t-doc__header,.dark-mode .t-doc__header{animation:headerbackground forwards;animation-timeline:scroll();animation-range:0px 200px}.dark-mode h2.t-editor__heading,.dark-mode .t-editor__page-title h1,.dark-mode h1.section-header,.dark-mode .markdown h1,.dark-mode .markdown h2,.dark-mode .markdown h3,.dark-mode .markdown h4,.dark-mode .markdown h5,.dark-mode .markdown h6{-webkit-text-fill-color:transparent;background-image:linear-gradient(to right bottom,rgb(255,255,255) 30%,rgba(255,255,255,.38));-webkit-background-clip:text;background-clip:text}.code-languages-background{background:var(--default-theme-background-1)!important}.code-languages-background:before{box-shadow:inset 0 0 0 2px var(--default-theme-border-color);background:linear-gradient(to right top,rgb(211 225 249 / 12%),rgb(209 223 247 / 11%),rgb(223 233 251 / 29%))!important}.light-mode .code-languages-background:before{background:linear-gradient(to right top,rgb(0 0 0 / 8%),rgb(0 0 0 / 7%),rgb(0 0 0 / 19%))!important}.code-languages__active .code-languages-background:before{background:var(--default-theme-background-1)!important}.sidebar-heading-type{color:var(--default-theme-background-1)!important}.active_page .sidebar-heading-type{color:var(--sidebar-color-1)!important}.sidebar-heading-type:after{content:"";position:absolute;top:0;left:0;width:100%;height:100%;background:linear-gradient(9deg,var(--default-theme-background-1),transparent);border-radius:30px;opacity:.3}.code-languages-icon{padding:12px!important}.code-languages span{margin-top:5px!important}.section-flare-item:nth-of-type(1),.section-flare-item:nth-of-type(2){--c1: #ffffff;--c2: #babfd8;--c3: #2e8bb2;--c4: #1a8593;--c5: #0a143e;--c6: #0a0f52;--c7: #2341b8;--solid: var(--c1), var(--c2), var(--c3), var(--c4), var(--c5), var(--c6), var(--c7);--solid-wrap: var(--solid), var(--c1);--trans: var(--c1), transparent, var(--c2), transparent, var(--c3), transparent, var(--c4), transparent, var(--c5), transparent, var(--c6), transparent, var(--c7);--trans-wrap: var(--trans), transparent, var(--c1);background:radial-gradient(circle,var(--trans)),conic-gradient(from 180deg,var(--trans-wrap)),radial-gradient(circle,var(--trans)),conic-gradient(var(--solid-wrap));width:600px;height:400px;border-radius:50%;filter:blur(30px);z-index:0;right:0;position:absolute;transform:rotate(-45deg);mix-blend-mode:hard-light}.section-flare-item:nth-of-type(2){transform:rotate(-45deg);width:100vw;top:-300px;position:absolute;height:1000px;filter:blur(100px);opacity:.3}.section-flare-item:nth-of-type(3){--star-color: #fff;--star-color2: #eee;--star-color3: #ddd;background-image:radial-gradient(2px 2px at 20px 30px,var(--star-color2),rgba(0,0,0,0)),radial-gradient(2px 2px at 40px 70px,var(--star-color),rgba(0,0,0,0)),radial-gradient(2px 2px at 50px 160px,var(--star-color3),rgba(0,0,0,0)),radial-gradient(2px 2px at 90px 40px,var(--star-color),rgba(0,0,0,0)),radial-gradient(2px 2px at 130px 80px,var(--star-color),rgba(0,0,0,0)),radial-gradient(2px 2px at 160px 120px,var(--star-color3),rgba(0,0,0,0));background-repeat:repeat;background-size:200px 200px;width:100%;height:100%;-webkit-mask-image:radial-gradient(ellipse at 100% 0%,black 40%,transparent 70%);mask-image:radial-gradient(ellipse at 100% 0%,black 40%,transparent 70%)}.section-flare{top:-150px!important;height:100vh;background:linear-gradient(#000,var(--default-theme-background-1));width:100vw}.light-mode .section-flare{background:linear-gradient(180deg,#f4fdff 30%,transparent 100%)}.light-mode .section-flare-item:nth-of-type(1){--c1: #ffffff;--c2: #eff0f5;--c3: #ffffff;--c4: #ffffff;--c5: #ffffff;--c6: #ffffff;--c7: #ebf2f5;filter:blur(5px);mix-blend-mode:initial}.light-mode .section-flare-item:nth-of-type(2){opacity:.1}
`, gr = `.light-mode{--default-theme-color-1: rgb(9, 9, 11);--default-theme-color-2: rgb(113, 113, 122);--default-theme-color-3: rgba(25, 25, 28, .5);--default-theme-color-accent: var(--default-theme-color-1);--default-theme-background-1: #fff;--default-theme-background-2: #f4f4f5;--default-theme-background-3: #e3e3e6;--default-theme-background-accent: #8ab4f81f;--default-theme-border-color: rgb(228, 228, 231);--default-theme-code-language-color-supersede: var(--default-theme-color-1)}.dark-mode{--default-theme-color-1: #fafafa;--default-theme-color-2: rgb(161, 161, 170);--default-theme-color-3: rgba(255, 255, 255, .533);--default-theme-color-accent: var(--default-theme-color-1);--default-theme-background-1: #09090b;--default-theme-background-2: #18181b;--default-theme-background-3: #2c2c30;--default-theme-background-accent: #8ab4f81f;--default-theme-border-color: rgba(255, 255, 255, .12);--default-theme-code-language-color-supersede: var(--default-theme-color-1)}.light-mode .t-doc__sidebar,.dark-mode .t-doc__sidebar{--sidebar-background-1: var(--default-theme-background-1);--sidebar-item-hover-color: currentColor;--sidebar-item-hover-background: var(--default-theme-background-2);--sidebar-item-active-background: var(--default-theme-background-3);--sidebar-border-color: var(--default-theme-border-color);--sidebar-color-1: var(--default-theme-color-1);--sidebar-color-2: var(--default-theme-color-2);--sidebar-color-active: var(--default-theme-color-accent);--sidebar-search-background: transparent;--sidebar-search-border-color: var(--default-theme-border-color);--sidebar-search--color: var(--default-theme-color-3)}.light-mode .t-doc__sidebar{--sidebar-item-active-background: #09090b;--sidebar-color-active: var(--sidebar-background-1)}.light-mode{--default-theme-button-1: rgb(49 53 56);--default-theme-button-1-color: #fff;--default-theme-button-1-hover: rgb(28 31 33);--default-theme-color-green: #069061;--default-theme-color-red: #ef0006;--default-theme-color-yellow: #edbe20;--default-theme-color-blue: #0082d0;--default-theme-color-orange: #fb892c;--default-theme-color-purple: #5203d1;--default-theme-scrollbar-color: rgba(0, 0, 0, .18);--default-theme-scrollbar-color-active: rgba(0, 0, 0, .36)}.dark-mode{--default-theme-button-1: #f6f6f6;--default-theme-button-1-color: #000;--default-theme-button-1-hover: #e7e7e7;--default-theme-color-green: rgba(69, 255, 165, .823);--default-theme-color-red: #ff8589;--default-theme-color-yellow: #ffcc4d;--default-theme-color-blue: #6bc1fe;--default-theme-color-orange: #f98943;--default-theme-color-purple: #b191f9;--default-theme-scrollbar-color: rgba(255, 255, 255, .24);--default-theme-scrollbar-color-active: rgba(255, 255, 255, .48)}.dark-mode h2.t-editor__heading,.dark-mode .t-editor__page-title h1,.dark-mode h1.section-header,.dark-mode .markdown h1,.dark-mode .markdown h2,.dark-mode .markdown h3,.dark-mode .markdown h4,.dark-mode .markdown h5,.dark-mode .markdown h6{-webkit-text-fill-color:transparent;background-image:linear-gradient(to right bottom,rgb(255,255,255) 30%,rgba(255,255,255,.38));-webkit-background-clip:text;background-clip:text}.light-mode .t-doc__header,.dark-mode .t-doc__header{animation:headerbackground forwards;animation-timeline:scroll();animation-range:0px 200px}.examples .scalar-card-footer{--default-theme-background-3: transparent;padding-top:0}.show-api-client-button:before{background:white!important}.show-api-client-button span,.show-api-client-button svg{color:#000!important}.download-cta,.references-rendered .markdown a{text-decoration:underline!important}@keyframes headerbackground{0%{background:transparent;-webkit-backdrop-filter:none;backdrop-filter:none}to{background:var(--header-background-1);-webkit-backdrop-filter:blur(12px);backdrop-filter:blur(12px)}}.code-languages-background{background:var(--default-theme-background-1)!important}.code-languages-background:before{box-shadow:inset 0 0 0 2px var(--default-theme-border-color);background:linear-gradient(to right top,rgb(211 225 249 / 12%),rgb(209 223 247 / 11%),rgb(223 233 251 / 29%))!important}.light-mode .code-languages-background:before{background:linear-gradient(to right top,rgb(0 0 0 / 8%),rgb(0 0 0 / 7%),rgb(0 0 0 / 19%))!important}.code-languages__active .code-languages-background:before{background:var(--default-theme-background-1)!important}.sidebar-heading-type:after{content:"";position:absolute;top:0;left:0;width:100%;height:100%;background:linear-gradient(9deg,var(--default-theme-background-1),transparent);border-radius:30px;opacity:.3}.code-languages-icon{padding:12px!important}.code-languages span{margin-top:5px!important}.section-flare{width:100vw;height:550px;position:relative}.section-flare-item:nth-of-type(1){position:absolute;width:100vw;height:550px;--stripesDark: repeating-linear-gradient( 100deg, #000 0%, #000 7%, transparent 10%, transparent 12%, #000 16% );--rainbow: repeating-linear-gradient( 100deg, #fff 10%, #fff 16%, #fff 22%, #fff 30% );background-image:var(--stripesDark),var(--rainbow);background-size:300%,200%;background-position:50% 50%,50% 50%;filter:invert(100%);-webkit-mask-image:radial-gradient(ellipse at 100% 0%,black 40%,transparent 70%);mask-image:radial-gradient(ellipse at 100% 0%,black 40%,transparent 70%);pointer-events:none;opacity:.07}.dark-mode .section-flare-item:nth-of-type(1){background-image:var(--stripesDark),var(--rainbow);filter:opacity(50%) saturate(200%);opacity:.25;height:350px}.section-flare-item:nth-of-type(1):after{content:"";position:absolute;top:0;right:0;bottom:0;left:0;background-image:var(--stripesDark),var(--rainbow);background-size:200%,100%;background-attachment:fixed;mix-blend-mode:difference}.dark-mode .section-flare:after{background-image:var(--stripesDark),var(--rainbow)}.section-flare-item:nth-of-type(2){--star-color: #fff;--star-color2: #fff;--star-color3: #fff;width:100%;height:100%;position:absolute;background-image:radial-gradient(2px 2px at 20px 30px,var(--star-color2),rgba(0,0,0,0)),radial-gradient(2px 2px at 40px 70px,var(--star-color),rgba(0,0,0,0)),radial-gradient(2px 2px at 50px 160px,var(--star-color3),rgba(0,0,0,0)),radial-gradient(2px 2px at 90px 40px,var(--star-color),rgba(0,0,0,0)),radial-gradient(2px 2px at 130px 80px,var(--star-color),rgba(0,0,0,0)),radial-gradient(2px 2px at 160px 120px,var(--star-color3),rgba(0,0,0,0));background-repeat:repeat;background-size:200px 200px;-webkit-mask-image:radial-gradient(ellipse at 100% 0%,black 40%,transparent 70%);mask-image:radial-gradient(ellipse at 100% 0%,black 40%,transparent 70%);opacity:.2}
`, wt = `.light-mode{color-scheme:light;--default-theme-color-1: #2a2f45;--default-theme-color-2: #757575;--default-theme-color-3: #8e8e8e;--default-theme-color-disabled: #b4b1b1;--default-theme-color-ghost: #a7a7a7;--default-theme-color-accent: #0099ff;--default-theme-background-1: #fff;--default-theme-background-2: #f6f6f6;--default-theme-background-3: #e7e7e7;--default-theme-background-4: rgba(0, 0, 0, .06);--default-theme-background-accent: #8ab4f81f;--default-theme-border-color: rgba(0, 0, 0, .1);--default-theme-scrollbar-color: rgba(0, 0, 0, .18);--default-theme-scrollbar-color-active: rgba(0, 0, 0, .36);--default-theme-lifted-brightness: 1;--default-theme-backdrop-brightness: 1;--default-theme-shadow-1: 0 1px 3px 0 rgba(0, 0, 0, .11);--default-theme-shadow-2: rgba(0, 0, 0, .08) 0px 13px 20px 0px, rgba(0, 0, 0, .08) 0px 3px 8px 0px, #eeeeed 0px 0 0 1px;--default-theme-button-1: rgb(49 53 56);--default-theme-button-1-color: #fff;--default-theme-button-1-hover: rgb(28 31 33);--default-theme-color-green: #069061;--default-theme-color-red: #ef0006;--default-theme-color-yellow: #edbe20;--default-theme-color-blue: #0082d0;--default-theme-color-orange: #fb892c;--default-theme-color-purple: #5203d1;--default-scalar-brand: #faf6e9}.dark-mode{color-scheme:dark;--default-theme-color-1: rgba(255, 255, 255, .9);--default-theme-color-2: rgba(255, 255, 255, .62);--default-theme-color-3: rgba(255, 255, 255, .44);--default-theme-color-disabled: rgba(255, 255, 255, .34);--default-theme-color-ghost: rgba(255, 255, 255, .26);--default-theme-color-accent: #8ab4f8;--default-theme-background-1: #1a1a1a;--default-theme-background-2: #252525;--default-theme-background-3: #323232;--default-theme-background-4: rgba(255, 255, 255, .06);--default-theme-background-accent: #8ab4f81f;--default-theme-border-color: rgba(255, 255, 255, .1);--default-theme-scrollbar-color: rgba(255, 255, 255, .24);--default-theme-scrollbar-color-active: rgba(255, 255, 255, .48);--default-theme-lifted-brightness: 1.45;--default-theme-backdrop-brightness: .5;--default-theme-shadow-1: 0 1px 3px 0 rgb(0, 0, 0, .1);--default-theme-shadow-2: rgba(15, 15, 15, .2) 0px 3px 6px, rgba(15, 15, 15, .4) 0px 9px 24px, 0 0 0 1px rgba(255, 255, 255, .1);--default-theme-button-1: #f6f6f6;--default-theme-button-1-color: #000;--default-theme-button-1-hover: #e7e7e7;--default-theme-color-green: #00b648;--default-theme-color-red: #dc1b19;--default-theme-color-yellow: #ffc90d;--default-theme-color-blue: #4eb3ec;--default-theme-color-orange: #ff8d4d;--default-theme-color-purple: #b191f9;--default-scalar-brand: #8b7256}.light-mode .t-doc__sidebar,.dark-mode .sidebar{--default-sidebar-background-1: var(--default-theme-background-1);--default-sidebar-item-hover-color: currentColor;--default-sidebar-item-hover-background: var(--default-theme-background-2);--default-sidebar-item-active-background: var( --default-theme-background-accent );--default-sidebar-border-color: var(--default-theme-border-color);--default-sidebar-color-1: var(--default-theme-color-1);--default-sidebar-color-2: var(--default-theme-color-2);--default-sidebar-color-active: var(--default-theme-color-accent);--default-sidebar-search-background: transparent;--default-sidebar-search-border-color: var(--default-theme-border-color);--default-sidebar-search--color: var(--default-theme-color-3)}
`, pr = `.light-mode{--default-theme-color-1: #2a2f45;--default-theme-color-2: #757575;--default-theme-color-3: #8e8e8e;--default-theme-color-accent: #7070ff;--default-theme-background-1: #fff;--default-theme-background-2: #f6f6f6;--default-theme-background-3: #e7e7e7;--default-theme-background-accent: #7070ff1f;--default-theme-border-color: rgba(0, 0, 0, .1);--default-theme-code-language-color-supersede: var(--default-theme-color-3)}.dark-mode{--default-theme-color-1: #f7f8f8;--default-theme-color-2: rgb(180, 188, 208);--default-theme-color-3: #b4bcd099;--default-theme-color-accent: #828fff;--default-theme-background-1: #000212;--default-theme-background-2: rgba(255, 255, 255, .05);--default-theme-background-3: rgba(255, 255, 255, .05);--default-theme-background-accent: #8ab4f81f;--default-theme-border-color: #242537;--default-theme-code-language-color-supersede: var(--default-theme-color-3)}.light-mode .t-doc__header{--header-background-1: rgba(255, 255, 255, .1);--header-border-color: var(--default-theme-border-color);--header-color-1: var(--default-theme-color-1);--header-color-2: var(--default-theme-color-2);--header-background-toggle: var(--default-theme-color-3);--header-call-to-action-color: var(--default-theme-color-accent);-webkit-backdrop-filter:blur(30px);backdrop-filter:blur(30px)}.dark-mode .t-doc__header{--header-background-1: rgba(0, 2, 18, .1);--header-border-color: var(--default-theme-border-color);--header-color-1: var(--default-theme-color-1);--header-color-2: var(--default-theme-color-2);--header-background-toggle: var(--default-theme-color-3);--header-call-to-action-color: var(--default-theme-color-accent);-webkit-backdrop-filter:blur(20px);backdrop-filter:blur(20px)}.light-mode .t-doc__sidebar{--sidebar-background-1: var(--default-theme-background-1);--sidebar-item-hover-color: currentColor;--sidebar-item-hover-background: var(--default-theme-background-2);--sidebar-item-active-background: var(--default-theme-background-accent);--sidebar-border-color: var(--default-theme-border-color);--sidebar-color-1: var(--default-theme-color-1);--sidebar-color-2: var(--default-theme-color-2);--sidebar-color-active: var(--default-theme-color-accent);--sidebar-search-background: rgba(0, 0, 0, .05);--sidebar-search-border-color: 1px solid rgba(0, 0, 0, .05);--sidebar-search--color: var(--default-theme-color-3);--default-theme-background-2: rgba(0, 0, 0, .03)}.dark-mode .t-doc__sidebar{--sidebar-background-1: var(--default-theme-background-1);--sidebar-item-hover-color: currentColor;--sidebar-item-hover-background: var(--default-theme-background-2);--sidebar-item-active-background: rgba(255, 255, 255, .1);--sidebar-border-color: var(--default-theme-border-color);--sidebar-color-1: var(--default-theme-color-1);--sidebar-color-2: var(--default-theme-color-2);--sidebar-color-active: var(--default-theme-color-accent);--sidebar-search-background: rgba(255, 255, 255, .1);--sidebar-search-border-color: 1px solid rgba(255, 255, 255, .05);--sidebar-search--color: var(--default-theme-color-3)}.light-mode{--default-theme-button-1: rgb(49 53 56);--default-theme-button-1-color: #fff;--default-theme-button-1-hover: rgb(28 31 33);--default-theme-color-green: #069061;--default-theme-color-red: #ef0006;--default-theme-color-yellow: #edbe20;--default-theme-color-blue: #0082d0;--default-theme-color-orange: #fb892c;--default-theme-color-purple: #5203d1;--default-theme-scrollbar-color: rgba(0, 0, 0, .18);--default-theme-scrollbar-color-active: rgba(0, 0, 0, .36)}.dark-mode{--default-theme-button-1: #f6f6f6;--default-theme-button-1-color: #000;--default-theme-button-1-hover: #e7e7e7;--default-theme-color-green: #00b648;--default-theme-color-red: #dc1b19;--default-theme-color-yellow: #ffc90d;--default-theme-color-blue: #4eb3ec;--default-theme-color-orange: #ff8d4d;--default-theme-color-purple: #b191f9;--default-theme-scrollbar-color: rgba(255, 255, 255, .24);--default-theme-scrollbar-color-active: rgba(255, 255, 255, .48)}.dark-mode h2.t-editor__heading,.dark-mode .t-editor__page-title h1,.dark-mode h1.section-header,.dark-mode .markdown h1,.dark-mode .markdown h2,.dark-mode .markdown h3,.dark-mode .markdown h4,.dark-mode .markdown h5,.dark-mode .markdown h6{-webkit-text-fill-color:transparent;background-image:linear-gradient(to right bottom,rgb(255,255,255) 30%,rgba(255,255,255,.38));-webkit-background-clip:text;background-clip:text}.sidebar-search{-webkit-backdrop-filter:blur(12px);backdrop-filter:blur(12px)}@keyframes headerbackground{0%{background:transparent;-webkit-backdrop-filter:none;backdrop-filter:none}to{background:var(--header-background-1);-webkit-backdrop-filter:blur(12px);backdrop-filter:blur(12px)}}.dark-mode .scalar-card{background:rgba(255,255,255,.05)!important}.dark-mode .scalar-card *{--default-theme-background-2: transparent !important;--default-theme-background-1: transparent !important}.light-mode .dark-mode.scalar-card *,.light-mode .dark-mode.scalar-card{--default-theme-background-1: #0d0f1e !important;--default-theme-background-2: #0d0f1e !important;--default-theme-background-3: #191b29 !important}.light-mode .dark-mode.scalar-card{background:#191b29!important}.sidebar-heading-type:before{z-index:10;position:relative}.sidebar-heading-type:after{content:"";position:absolute;top:0;left:0;width:100%;height:100%;background:linear-gradient(9deg,var(--default-theme-background-1),transparent);border-radius:30px;opacity:.3}.show-api-client-button:before{background-color:var(--default-theme-color-accent)!important}.badge{box-shadow:0 0 0 1px var(--default-theme-border-color);margin-right:6px}.table-row.required-parameter .table-row-item:nth-of-type(2):after{background:transparent;box-shadow:none}.code-languages__active{--default-theme-code-language-color-supersede: var(--default-theme-color-1)}.code-languages-background{background:linear-gradient(rgba(255,255,255,0) 0%,rgba(255,255,255,.05) 100%)!important;box-shadow:inset 0 0 0 1px var(--default-theme-border-color)!important}.code-languages-icon{padding:12px!important}.section-flare{width:100vw;background:radial-gradient(ellipse 80% 50% at 50% -20%,rgba(120,119,198,.3),transparent);height:100%}
`, kr = `.light-mode{--default-theme-color-1: #2a2f45;--default-theme-color-2: #757575;--default-theme-color-3: #8e8e8e;--default-theme-color-accent: #ff0000;--default-theme-background-1: #ffecdf;--default-theme-background-2: #f5e0d2;--default-theme-background-3: #e1caba;--default-theme-background-accent: #dcbfa81f;--default-theme-border-color: rgba(0, 0, 0, .1);--default-theme-code-language-color-supersede: var(--default-theme-color-1)}.dark-mode{--default-theme-color-1: rgba(255, 255, 255, .9);--default-theme-color-2: rgba(255, 255, 255, .62);--default-theme-color-3: rgba(255, 255, 255, .44);--default-theme-color-accent: #eca78e;--default-theme-background-1: #140507;--default-theme-background-2: #20090c;--default-theme-background-3: #321116;--default-theme-background-accent: #441313;--default-theme-border-color: rgba(255, 255, 255, .1);--default-theme-code-language-color-supersede: var(--default-theme-color-1)}.light-mode .t-doc__sidebar,.dark-mode .t-doc__sidebar{--sidebar-background-1: transparent;--sidebar-item-hover-color: currentColor;--sidebar-item-hover-background: var(--default-theme-background-2);--sidebar-item-active-background: var(--default-theme-background-3);--sidebar-border-color: var(--default-theme-border-color);--sidebar-color-1: var(--default-theme-color-1);--sidebar-color-2: var(--default-theme-color-2);--sidebar-color-active: var(--default-theme-color-accent);--sidebar-search-background: rgba(255, 255, 255, .1);--sidebar-search-border-color: var(--default-theme-border-color);--sidebar-search--color: var(--default-theme-color-3);z-index:1}.light-mode{--default-theme-button-1: rgb(49 53 56);--default-theme-button-1-color: #fff;--default-theme-button-1-hover: rgb(28 31 33);--default-theme-color-green: #069061;--default-theme-color-red: #ef0006;--default-theme-color-yellow: #edbe20;--default-theme-color-blue: #0082d0;--default-theme-color-orange: #fb892c;--default-theme-color-purple: #5203d1;--default-theme-scrollbar-color: rgba(0, 0, 0, .18);--default-theme-scrollbar-color-active: rgba(0, 0, 0, .36)}.dark-mode{--default-theme-button-1: #f6f6f6;--default-theme-button-1-color: #000;--default-theme-button-1-hover: #e7e7e7;--default-theme-color-green: rgba(69, 255, 165, .823);--default-theme-color-red: #ff8589;--default-theme-color-yellow: #ffcc4d;--default-theme-color-blue: #6bc1fe;--default-theme-color-orange: #f98943;--default-theme-color-purple: #b191f9;--default-theme-scrollbar-color: rgba(255, 255, 255, .24);--default-theme-scrollbar-color-active: rgba(255, 255, 255, .48)}.dark-mode h2.t-editor__heading,.dark-mode .t-editor__page-title h1,.dark-mode h1.section-header,.dark-mode .markdown h1,.dark-mode .markdown h2,.dark-mode .markdown h3,.dark-mode .markdown h4,.dark-mode .markdown h5,.dark-mode .markdown h6{-webkit-text-fill-color:transparent;background-image:linear-gradient(to right bottom,rgb(255,255,255) 30%,rgba(255,255,255,.38));-webkit-background-clip:text;background-clip:text}.light-mode .t-doc__sidebar{--sidebar-search-background: white}.examples .scalar-card-footer{--default-theme-background-3: transparent;padding-top:0}.show-api-client-button:before{background:white!important}.show-api-client-button span,.show-api-client-button svg{color:#000!important}@keyframes headerbackground{0%{background:transparent;-webkit-backdrop-filter:none;backdrop-filter:none}to{background:var(--header-background-1);-webkit-backdrop-filter:blur(12px);backdrop-filter:blur(12px)}}.light-mode .t-doc__header,.dark-mode .t-doc__header{animation:headerbackground forwards;animation-timeline:scroll();animation-range:0px 200px}.code-languages-background{background:var(--default-theme-background-1)!important}.code-languages-background:before{box-shadow:inset 0 0 0 2px var(--default-theme-border-color);background:linear-gradient(to right top,rgb(211 225 249 / 12%),rgb(209 223 247 / 11%),rgb(223 233 251 / 29%))!important}.light-mode .code-languages-background:before{background:linear-gradient(to right top,rgb(0 0 0 / 8%),rgb(0 0 0 / 7%),rgb(0 0 0 / 19%))!important}.code-languages__active .code-languages-background:before{background:var(--default-theme-background-1)!important}.sidebar-heading-type{color:var(--default-theme-background-1)!important}.active_page .sidebar-heading-type{color:var(--sidebar-color-active)!important}.sidebar-heading-type:after{content:"";position:absolute;top:0;left:0;width:100%;height:100%;background:linear-gradient(9deg,var(--default-theme-background-1),transparent);border-radius:30px;opacity:.3}.code-languages-icon{padding:12px!important}.code-languages span{margin-top:5px!important}.section-flare-item:nth-of-type(1){--c1: #f3a041;--c2: #eb7b42;--c3: #b03f2e;--c4: #95281a;--c5: #3f0f0a;--c6: #53120a;--c7: #b63923;--solid: var(--c1), var(--c2), var(--c3), var(--c4), var(--c5), var(--c6), var(--c7);--solid-wrap: var(--solid), var(--c1);--trans: var(--c1), transparent, var(--c2), transparent, var(--c3), transparent, var(--c4), transparent, var(--c5), transparent, var(--c6), transparent, var(--c7);--trans-wrap: var(--trans), transparent, var(--c1);background:radial-gradient(circle,var(--trans)),conic-gradient(from 180deg,var(--trans-wrap)),radial-gradient(circle,var(--trans)),conic-gradient(var(--solid-wrap));width:100vw;height:600px;margin-top:-150px;border-radius:50%;filter:blur(30px);z-index:0}.light-mode .section-flare-item:nth-of-type(1){opacity:.15;height:600px;margin-top:-150px;mix-blend-mode:luminosity}.section-flare{top:-150px!important;height:100vh;background:linear-gradient(#e63a0a,var(--default-theme-background-1))}.light-mode .section-flare{background:transparent}
`, vr = `.light-mode{color-scheme:light;--default-theme-color-1: #000000;--default-theme-color-2: #000000;--default-theme-color-3: #000000;--default-theme-color-accent: #645b0f;--default-theme-background-1: #ccc9b3;--default-theme-background-2: #c2bfaa;--default-theme-background-3: #b8b5a1;--default-theme-background-accent: #000000;--default-theme-border-color: rgba(0, 0, 0, .2);--default-theme-scrollbar-color: rgba(0, 0, 0, .18);--default-theme-scrollbar-color-active: rgba(0, 0, 0, .36);--default-theme-lifted-brightness: 1;--default-theme-backdrop-brightness: 1;--default-theme-shadow-1: 0 1px 3px 0 rgba(0, 0, 0, .11);--default-theme-shadow-2: rgba(0, 0, 0, .08) 0px 13px 20px 0px, rgba(0, 0, 0, .08) 0px 3px 8px 0px, var(--default-theme-border-color) 0px 0 0 1px;--default-theme-button-1: rgb(49 53 56);--default-theme-button-1-color: #fff;--default-theme-button-1-hover: rgb(28 31 33);--default-theme-color-red: #b91c1c;--default-theme-color-orange: #a16207;--default-theme-color-green: #047857;--default-theme-color-blue: #1d4ed8;--default-theme-color-orange: #c2410c;--default-theme-color-purple: #6d28d9;--default-theme-code-languages-background-supersede: var( --default-theme-background-3 );--default-theme-code-language-color-supersede: var(--default-theme-color-1)}.dark-mode{color-scheme:dark;--default-theme-color-1: #fffef3;--default-theme-color-2: #fffef3;--default-theme-color-3: #fffef3;--default-theme-color-accent: #c3b531;--default-theme-background-1: #313332;--default-theme-background-2: #393b3a;--default-theme-background-3: #414342;--default-theme-background-accent: #fffef3;--default-theme-border-color: rgba(255, 255, 255, .1);--default-theme-scrollbar-color: rgba(255, 255, 255, .24);--default-theme-scrollbar-color-active: rgba(255, 255, 255, .48);--default-theme-lifted-brightness: 1.45;--default-theme-backdrop-brightness: .5;--default-theme-shadow-1: 0 1px 3px 0 rgba(0, 0, 0, .11);--default-theme-shadow-2: rgba(15, 15, 15, .2) 0px 3px 6px, rgba(15, 15, 15, .4) 0px 9px 24px, 0 0 0 1px rgba(255, 255, 255, .1);--default-theme-button-1: #f6f6f6;--default-theme-button-1-color: #000;--default-theme-button-1-hover: #e7e7e7;--default-theme-color-green: #00b648;--default-theme-color-red: #dc1b19;--default-theme-color-yellow: #ffc90d;--default-theme-color-blue: #4eb3ec;--default-theme-color-orange: #ff8d4d;--default-theme-color-purple: #b191f9;--default-theme-code-languages-background-supersede: var( --default-theme-background-3 );--default-theme-code-language-color-supersede: var(--default-theme-color-1)}.light-mode .t-doc__sidebar,.dark-mode .sidebar{--default-sidebar-background-1: var(--default-theme-background-1);--default-sidebar-item-hover-color: currentColor;--default-sidebar-item-hover-background: var(--default-theme-background-2);--default-sidebar-item-active-background: var( --default-theme-background-accent );--default-sidebar-border-color: var(--default-theme-border-color);--default-sidebar-color-1: var(--default-theme-color-1);--default-sidebar-color-2: var(--default-theme-color-2);--default-sidebar-color-active: var(--default-sidebar-background-1);--default-sidebar-search-background: var(--default-theme-background-3);--default-sidebar-search-border-color: var( --default-sidebar-search-background );--default-sidebar-search--color: var(--default-theme-color-3)}
`, xr = `.light-mode{color-scheme:light;--default-theme-color-1: #24293c;--default-theme-color-2: #4a5166;--default-theme-color-3: #888c9a;--default-theme-color-accent: #5369d2;--default-theme-background-1: #fff;--default-theme-background-2: #f6f6f6;--default-theme-background-3: #e7e7e7;--default-theme-background-accent: #5369d20f;--default-theme-border-color: #d8e0e8;--default-theme-scrollbar-color: rgba(0, 0, 0, .18);--default-theme-scrollbar-color-active: rgba(0, 0, 0, .36);--default-theme-lifted-brightness: 1;--default-theme-backdrop-brightness: 1;--default-theme-shadow-1: 0 1px 3px 0 rgba(0, 0, 0, .11);--default-theme-shadow-2: rgba(0, 0, 0, .08) 0px 13px 20px 0px, rgba(0, 0, 0, .08) 0px 3px 8px 0px, var(--default-theme-border-color) 0px 0 0 1px;--default-theme-button-1: rgb(49 53 56);--default-theme-button-1-color: #fff;--default-theme-button-1-hover: rgb(28 31 33);--default-theme-color-green: #069061;--default-theme-color-red: #ef0006;--default-theme-color-yellow: #edbe20;--default-theme-color-blue: #0082d0;--default-theme-color-orange: #fb892c;--default-theme-color-purple: #5203d1}.dark-mode{color-scheme:dark;--default-theme-color-1: #f2f2f4;--default-theme-color-2: #acaeb6;--default-theme-color-3: #858893;--default-theme-color-accent: #5579df;--default-theme-background-1: #262a3f;--default-theme-background-2: #383c4f;--default-theme-background-3: #4a4d5f;--default-theme-background-accent: #5579df1f;--default-theme-border-color: rgba(255, 255, 255, .1);--default-theme-scrollbar-color: rgba(255, 255, 255, .24);--default-theme-scrollbar-color-active: rgba(255, 255, 255, .48);--default-theme-lifted-brightness: 1.45;--default-theme-backdrop-brightness: .5;--default-theme-shadow-1: 0 1px 3px 0 rgb(0, 0, 0, .1);--default-theme-shadow-2: rgba(15, 15, 15, .2) 0px 3px 6px, rgba(15, 15, 15, .4) 0px 9px 24px, 0 0 0 1px rgba(255, 255, 255, .1);--default-theme-button-1: #f6f6f6;--default-theme-button-1-color: #000;--default-theme-button-1-hover: #e7e7e7;--default-theme-color-green: #00b648;--default-theme-color-red: #dc1b19;--default-theme-color-yellow: #ffc90d;--default-theme-color-blue: #4eb3ec;--default-theme-color-orange: #ff8d4d;--default-theme-color-purple: #b191f9}.light-mode .t-doc__sidebar,.dark-mode .sidebar{--default-sidebar-background-1: var(--default-theme-background-1);--default-sidebar-item-hover-color: currentColor;--default-sidebar-item-hover-background: var(--default-theme-background-2);--default-sidebar-item-active-background: var( --default-theme-background-accent );--default-sidebar-border-color: var(--default-theme-border-color);--default-sidebar-color-1: var(--default-theme-color-1);--default-sidebar-color-2: var(--default-theme-color-2);--default-sidebar-color-active: var(--default-theme-color-accent);--default-sidebar-search-background: var(--default-theme-background-2);--default-sidebar-search-border-color: var(--default-theme-border-color);--default-sidebar-search--color: var(--default-theme-color-3)}
`, _r = `.light-mode{--default-theme-color-1: #2a2f45;--default-theme-color-2: #757575;--default-theme-color-3: #8e8e8e;--default-theme-color-accent: #1f808d;--default-theme-background-1: #fbf6e5;--default-theme-background-2: #efead7;--default-theme-background-3: #e0dbc7;--default-theme-background-accent: #1f808d1f;--default-theme-border-color: rgba(215, 215, 206, .5);--default-theme-code-language-color-supersede: var(--default-theme-color-1)}.dark-mode{--default-theme-color-1: rgba(255, 255, 255, .9);--default-theme-color-2: rgba(255, 255, 255, .62);--default-theme-color-3: rgba(255, 255, 255, .44);--default-theme-color-accent: #ceb797;--default-theme-background-1: #1c1813;--default-theme-background-2: #29241f;--default-theme-background-3: #36302b;--default-theme-background-accent: #8ab4f81f;--default-theme-border-color: rgba(255, 255, 255, .1);--default-theme-code-language-color-supersede: var(--default-theme-color-1)}.light-mode .t-doc__header{--header-background-1: rgba(255, 255, 255, .1);--header-border-color: var(--default-theme-border-color);--header-color-1: var(--default-theme-color-1);--header-color-2: var(--default-theme-color-2);--header-background-toggle: var(--default-theme-color-3);--header-call-to-action-color: var(--default-theme-color-accent);-webkit-backdrop-filter:blur(30px);backdrop-filter:blur(30px)}.dark-mode .t-doc__header{--header-background-1: rgba(0, 2, 18, .1);--header-border-color: var(--default-theme-border-color);--header-color-1: var(--default-theme-color-1);--header-color-2: var(--default-theme-color-2);--header-background-toggle: var(--default-theme-color-3);--header-call-to-action-color: var(--default-theme-color-accent);-webkit-backdrop-filter:blur(20px);backdrop-filter:blur(20px)}.light-mode .t-doc__sidebar,.dark-mode .t-doc__sidebar{--sidebar-background-1: transparent;--sidebar-item-hover-color: currentColor;--sidebar-item-hover-background: var(--default-theme-background-3);--sidebar-item-active-background: var(--default-theme-background-3);--sidebar-border-color: var(--default-theme-background-2);--sidebar-color-1: var(--default-theme-color-1);--sidebar-color-2: var(--default-theme-color-2);--sidebar-color-active: var(--default-theme-color-1);--sidebar-search-background: var(--default-theme-background-1);--sidebar-search-border-color: var(--default-theme-border-color);--sidebar-search--color: var(--default-theme-color-3);z-index:10}.light-mode{--default-theme-button-1: rgb(49 53 56);--default-theme-button-1-color: #fff;--default-theme-button-1-hover: rgb(28 31 33);--default-theme-color-green: #17803d;--default-theme-color-red: #e10909;--default-theme-color-yellow: #edbe20;--default-theme-color-blue: #1763a6;--default-theme-color-orange: #e25b09;--default-theme-color-purple: #5c3993;--default-theme-scrollbar-color: rgba(0, 0, 0, .18);--default-theme-scrollbar-color-active: rgba(0, 0, 0, .36)}.dark-mode{--default-theme-button-1: #f6f6f6;--default-theme-button-1-color: #000;--default-theme-button-1-hover: #e7e7e7;--default-theme-color-green: #00b648;--default-theme-color-red: #dc1b19;--default-theme-color-yellow: #ffc90d;--default-theme-color-blue: #4eb3ec;--default-theme-color-orange: #ff8d4d;--default-theme-color-purple: #b191f9;--default-theme-scrollbar-color: rgba(255, 255, 255, .24);--default-theme-scrollbar-color-active: rgba(255, 255, 255, .48)}.dark-mode h2.t-editor__heading,.dark-mode .t-editor__page-title h1,.dark-mode h1.section-header,.dark-mode .markdown h1,.dark-mode .markdown h2,.dark-mode .markdown h3,.dark-mode .markdown h4,.dark-mode .markdown h5,.dark-mode .markdown h6{-webkit-text-fill-color:transparent;background-image:linear-gradient(to right bottom,rgb(255,255,255) 30%,rgba(255,255,255,.38));-webkit-background-clip:text;background-clip:text}.t-doc__sidebar .sidebar-heading-type{color:var(--default-theme-background-1)}.code-languages-background{background:var(--default-theme-background-1)!important}.scalar-card .code-languages-icon{padding:12px}.code-languages-background:before{box-shadow:inset 0 0 0 2px var(--default-theme-border-color);background:linear-gradient(to right top,rgb(211 225 249 / 12%),rgb(209 223 247 / 11%),rgb(223 233 251 / 29%))!important}.light-mode .code-languages-background:before{background:linear-gradient(to right top,rgb(0 0 0 / 8%),rgb(0 0 0 / 7%),rgb(0 0 0 / 19%))!important}.code-languages__active .code-languages-background:before{background:var(--default-theme-background-1)!important}@keyframes headerbackground{0%{background:transparent;-webkit-backdrop-filter:none;backdrop-filter:none}to{background:var(--header-background-1);-webkit-backdrop-filter:blur(12px);backdrop-filter:blur(12px)}}.section-flare{height:100vh;width:100vw;position:absolute;right:0;top:-150px!important;overflow:hidden;z-index:0;filter:sepia(1) blur(12px);opacity:.4}.light-mode .section-flare{opacity:.45}.section-flare-item:nth-of-type(1),.section-flare-item:nth-of-type(2),.section-flare-item:nth-of-type(3){width:115%;min-width:1100px;height:70%;top:-250px;left:150px;background:radial-gradient(transparent 47%,rgba(255,255,255,.1) 48%,rgb(239 206 91) 50%,rgb(193 94 94 / 15%) 55%,transparent,#b1c3a8 60%,#272d2b8c 64%,transparent 64%,#ffffff21,#ffffff,transparent 80%,transparent);position:absolute;overflow:hidden;border-radius:50%}.section-flare-item:nth-of-type(2){transform:rotate(-16deg);left:-150px;width:200%;opacity:.1}.section-flare-item:nth-of-type(3){transform:rotate(16deg);left:-90%;width:250%;opacity:.1}
`, wr = `.light-mode{color-scheme:light;--default-theme-color-1: #584c27;--default-theme-color-2: #616161;--default-theme-color-3: #a89f84;--default-theme-color-accent: #b58900;--default-theme-background-1: #fdf6e3;--default-theme-background-2: #eee8d5;--default-theme-background-3: #ddd6c1;--default-theme-background-accent: #b589001f;--default-theme-border-color: #ded8c8;--default-theme-scrollbar-color: rgba(0, 0, 0, .18);--default-theme-scrollbar-color-active: rgba(0, 0, 0, .36);--default-theme-lifted-brightness: 1;--default-theme-backdrop-brightness: 1;--default-theme-shadow-1: 0 1px 3px 0 rgba(0, 0, 0, .11);--default-theme-shadow-2: rgba(0, 0, 0, .08) 0px 13px 20px 0px, rgba(0, 0, 0, .08) 0px 3px 8px 0px, #eeeeed 0px 0 0 1px;--default-theme-button-1: rgb(49 53 56);--default-theme-button-1-color: #fff;--default-theme-button-1-hover: rgb(28 31 33);--default-theme-color-red: #b91c1c;--default-theme-color-orange: #a16207;--default-theme-color-green: #047857;--default-theme-color-blue: #1d4ed8;--default-theme-color-orange: #c2410c;--default-theme-color-purple: #6d28d9}.dark-mode{color-scheme:dark;--default-theme-color-1: #fff;--default-theme-color-2: #cccccc;--default-theme-color-3: #6d8890;--default-theme-color-accent: #007acc;--default-theme-background-1: #00212b;--default-theme-background-2: #012b36;--default-theme-background-3: #004052;--default-theme-background-accent: #015a6f;--default-theme-border-color: rgba(255, 255, 255, .1);--default-theme-scrollbar-color: rgba(255, 255, 255, .24);--default-theme-scrollbar-color-active: rgba(255, 255, 255, .48);--default-theme-lifted-brightness: 1.45;--default-theme-backdrop-brightness: .5;--default-theme-shadow-1: 0 1px 3px 0 rgb(0, 0, 0, .1);--default-theme-shadow-2: rgba(15, 15, 15, .2) 0px 3px 6px, rgba(15, 15, 15, .4) 0px 9px 24px, 0 0 0 1px rgba(255, 255, 255, .1);--default-theme-button-1: #f6f6f6;--default-theme-button-1-color: #000;--default-theme-button-1-hover: #e7e7e7;--default-theme-color-green: #00b648;--default-theme-color-red: #dc1b19;--default-theme-color-yellow: #ffc90d;--default-theme-color-blue: #4eb3ec;--default-theme-color-orange: #ff8d4d;--default-theme-color-purple: #b191f9}.light-mode .t-doc__sidebar{--default-sidebar-background-1: var(--default-theme-background-1);--default-sidebar-item-hover-color: currentColor;--default-sidebar-item-hover-background: var(--default-theme-background-2);--default-sidebar-item-active-background: var( --default-theme-background-accent );--default-sidebar-border-color: var(--default-theme-border-color);--default-sidebar-color-1: var(--default-theme-color-1);--default-sidebar-color-2: var(--default-theme-color-2);--default-sidebar-color-active: var(--default-theme-color-accent);--default-sidebar-search-background: var(--default-theme-background-2);--default-sidebar-search-border-color: var(--sidebar-search-background);--default-sidebar-search--color: var(--default-theme-color-3)}.dark-mode .sidebar{--default-sidebar-background-1: var(--default-theme-background-1);--default-sidebar-item-hover-color: currentColor;--default-sidebar-item-hover-background: var(--default-theme-background-2);--default-sidebar-item-active-background: var( --default-theme-background-accent );--default-sidebar-border-color: var(--default-theme-border-color);--default-sidebar-color-1: var(--default-theme-color-1);--default-sidebar-color-2: var(--default-theme-color-2);--default-sidebar-color-active: var(--default-sidebar-color-1);--default-sidebar-search-background: var(--default-theme-background-2);--default-sidebar-search-border-color: var(--sidebar-search-background);--default-sidebar-search--color: var(--default-theme-color-3)}
`;
function Er(e, t) {
  const r = /* @__PURE__ */ Object.create(null), a = e.split(",");
  for (let o = 0; o < a.length; o++)
    r[a[o]] = !0;
  return t ? (o) => !!r[o.toLowerCase()] : (o) => !!r[o];
}
const $ = process.env.NODE_ENV !== "production" ? Object.freeze({}) : {}, Nr = process.env.NODE_ENV !== "production" ? Object.freeze([]) : [], Et = () => {
}, yr = /^on[^a-z]/, Or = (e) => yr.test(e), C = Object.assign, Sr = (e, t) => {
  const r = e.indexOf(t);
  r > -1 && e.splice(r, 1);
}, Cr = Object.prototype.hasOwnProperty, p = (e, t) => Cr.call(e, t), b = Array.isArray, K = (e) => Ee(e) === "[object Map]", Nt = (e) => Ee(e) === "[object Set]", v = (e) => typeof e == "function", S = (e) => typeof e == "string", je = (e) => typeof e == "symbol", x = (e) => e !== null && typeof e == "object", Vr = (e) => x(e) && v(e.then) && v(e.catch), yt = Object.prototype.toString, Ee = (e) => yt.call(e), Ot = (e) => Ee(e).slice(8, -1), St = (e) => Ee(e) === "[object Object]", Ke = (e) => S(e) && e !== "NaN" && e[0] !== "-" && "" + parseInt(e, 10) === e, Ct = (e) => {
  const t = /* @__PURE__ */ Object.create(null);
  return (r) => t[r] || (t[r] = e(r));
}, Tr = /-(\w)/g, ge = Ct((e) => e.replace(Tr, (t, r) => r ? r.toUpperCase() : "")), pe = Ct(
  (e) => e.charAt(0).toUpperCase() + e.slice(1)
), ke = (e, t) => !Object.is(e, t), Dr = (e, t, r) => {
  Object.defineProperty(e, t, {
    configurable: !0,
    enumerable: !1,
    value: r
  });
};
let rt;
const Ce = () => rt || (rt = typeof globalThis < "u" ? globalThis : typeof self < "u" ? self : typeof window < "u" ? window : typeof global < "u" ? global : {});
function He(e) {
  if (b(e)) {
    const t = {};
    for (let r = 0; r < e.length; r++) {
      const a = e[r], o = S(a) ? Pr(a) : He(a);
      if (o)
        for (const n in o)
          t[n] = o[n];
    }
    return t;
  } else {
    if (S(e))
      return e;
    if (x(e))
      return e;
  }
}
const Ir = /;(?![^(]*\))/g, Rr = /:([^]+)/, $r = /\/\*[^]*?\*\//g;
function Pr(e) {
  const t = {};
  return e.replace($r, "").split(Ir).forEach((r) => {
    if (r) {
      const a = r.split(Rr);
      a.length > 1 && (t[a[0].trim()] = a[1].trim());
    }
  }), t;
}
function Ue(e) {
  let t = "";
  if (S(e))
    t = e;
  else if (b(e))
    for (let r = 0; r < e.length; r++) {
      const a = Ue(e[r]);
      a && (t += a + " ");
    }
  else if (x(e))
    for (const r in e)
      e[r] && (t += r + " ");
  return t.trim();
}
const Mr = (e) => S(e) ? e : e == null ? "" : b(e) || x(e) && (e.toString === yt || !v(e.toString)) ? JSON.stringify(e, Vt, 2) : String(e), Vt = (e, t) => t && t.__v_isRef ? Vt(e, t.value) : K(t) ? {
  [`Map(${t.size})`]: [...t.entries()].reduce((r, [a, o]) => (r[`${a} =>`] = o, r), {})
} : Nt(t) ? {
  [`Set(${t.size})`]: [...t.values()]
} : x(t) && !b(t) && !St(t) ? String(t) : t;
function ot(e, ...t) {
  console.warn(`[Vue warn] ${e}`, ...t);
}
let Tt;
function zr(e, t = Tt) {
  t && t.active && t.effects.push(e);
}
function Fr() {
  return Tt;
}
const Ve = (e) => {
  const t = new Set(e);
  return t.w = 0, t.n = 0, t;
}, Dt = (e) => (e.w & F) > 0, It = (e) => (e.n & F) > 0, Ar = ({ deps: e }) => {
  if (e.length)
    for (let t = 0; t < e.length; t++)
      e[t].w |= F;
}, jr = (e) => {
  const { deps: t } = e;
  if (t.length) {
    let r = 0;
    for (let a = 0; a < t.length; a++) {
      const o = t[a];
      Dt(o) && !It(o) ? o.delete(e) : t[r++] = o, o.w &= ~F, o.n &= ~F;
    }
    t.length = r;
  }
}, Te = /* @__PURE__ */ new WeakMap();
let Z = 0, F = 1;
const De = 30;
let w;
const H = Symbol(process.env.NODE_ENV !== "production" ? "iterate" : ""), Ie = Symbol(process.env.NODE_ENV !== "production" ? "Map key iterate" : "");
class Kr {
  constructor(t, r = null, a) {
    this.fn = t, this.scheduler = r, this.active = !0, this.deps = [], this.parent = void 0, zr(this, a);
  }
  run() {
    if (!this.active)
      return this.fn();
    let t = w, r = U;
    for (; t; ) {
      if (t === this)
        return;
      t = t.parent;
    }
    try {
      return this.parent = w, w = this, U = !0, F = 1 << ++Z, Z <= De ? Ar(this) : at(this), this.fn();
    } finally {
      Z <= De && jr(this), F = 1 << --Z, w = this.parent, U = r, this.parent = void 0, this.deferStop && this.stop();
    }
  }
  stop() {
    w === this ? this.deferStop = !0 : this.active && (at(this), this.onStop && this.onStop(), this.active = !1);
  }
}
function at(e) {
  const { deps: t } = e;
  if (t.length) {
    for (let r = 0; r < t.length; r++)
      t[r].delete(e);
    t.length = 0;
  }
}
let U = !0;
const Rt = [];
function $t() {
  Rt.push(U), U = !1;
}
function Pt() {
  const e = Rt.pop();
  U = e === void 0 ? !0 : e;
}
function N(e, t, r) {
  if (U && w) {
    let a = Te.get(e);
    a || Te.set(e, a = /* @__PURE__ */ new Map());
    let o = a.get(r);
    o || a.set(r, o = Ve());
    const n = process.env.NODE_ENV !== "production" ? { effect: w, target: e, type: t, key: r } : void 0;
    Hr(o, n);
  }
}
function Hr(e, t) {
  let r = !1;
  Z <= De ? It(e) || (e.n |= F, r = !Dt(e)) : r = !e.has(w), r && (e.add(w), w.deps.push(e), process.env.NODE_ENV !== "production" && w.onTrack && w.onTrack(
    C(
      {
        effect: w
      },
      t
    )
  ));
}
function A(e, t, r, a, o, n) {
  const l = Te.get(e);
  if (!l)
    return;
  let c = [];
  if (t === "clear")
    c = [...l.values()];
  else if (r === "length" && b(e)) {
    const i = Number(a);
    l.forEach((h, d) => {
      (d === "length" || d >= i) && c.push(h);
    });
  } else
    switch (r !== void 0 && c.push(l.get(r)), t) {
      case "add":
        b(e) ? Ke(r) && c.push(l.get("length")) : (c.push(l.get(H)), K(e) && c.push(l.get(Ie)));
        break;
      case "delete":
        b(e) || (c.push(l.get(H)), K(e) && c.push(l.get(Ie)));
        break;
      case "set":
        K(e) && c.push(l.get(H));
        break;
    }
  const u = process.env.NODE_ENV !== "production" ? { target: e, type: t, key: r, newValue: a, oldValue: o, oldTarget: n } : void 0;
  if (c.length === 1)
    c[0] && (process.env.NODE_ENV !== "production" ? le(c[0], u) : le(c[0]));
  else {
    const i = [];
    for (const h of c)
      h && i.push(...h);
    process.env.NODE_ENV !== "production" ? le(Ve(i), u) : le(Ve(i));
  }
}
function le(e, t) {
  const r = b(e) ? e : [...e];
  for (const a of r)
    a.computed && nt(a, t);
  for (const a of r)
    a.computed || nt(a, t);
}
function nt(e, t) {
  (e !== w || e.allowRecurse) && (process.env.NODE_ENV !== "production" && e.onTrigger && e.onTrigger(C({ effect: e }, t)), e.scheduler ? e.scheduler() : e.run());
}
const Ur = /* @__PURE__ */ Er("__proto__,__v_isRef,__isVue"), Mt = new Set(
  /* @__PURE__ */ Object.getOwnPropertyNames(Symbol).filter((e) => e !== "arguments" && e !== "caller").map((e) => Symbol[e]).filter(je)
), Wr = /* @__PURE__ */ We(), Br = /* @__PURE__ */ We(!0), Jr = /* @__PURE__ */ We(!0, !0), lt = /* @__PURE__ */ qr();
function qr() {
  const e = {};
  return ["includes", "indexOf", "lastIndexOf"].forEach((t) => {
    e[t] = function(...r) {
      const a = f(this);
      for (let n = 0, l = this.length; n < l; n++)
        N(a, "get", n + "");
      const o = a[t](...r);
      return o === -1 || o === !1 ? a[t](...r.map(f)) : o;
    };
  }), ["push", "pop", "shift", "unshift", "splice"].forEach((t) => {
    e[t] = function(...r) {
      $t();
      const a = f(this)[t].apply(this, r);
      return Pt(), a;
    };
  }), e;
}
function Gr(e) {
  const t = f(this);
  return N(t, "has", e), t.hasOwnProperty(e);
}
function We(e = !1, t = !1) {
  return function(a, o, n) {
    if (o === "__v_isReactive")
      return !e;
    if (o === "__v_isReadonly")
      return e;
    if (o === "__v_isShallow")
      return t;
    if (o === "__v_raw" && n === (e ? t ? Kt : jt : t ? so : At).get(a))
      return a;
    const l = b(a);
    if (!e) {
      if (l && p(lt, o))
        return Reflect.get(lt, o, n);
      if (o === "hasOwnProperty")
        return Gr;
    }
    const c = Reflect.get(a, o, n);
    return (je(o) ? Mt.has(o) : Ur(o)) || (e || N(a, "get", o), t) ? c : E(c) ? l && Ke(o) ? c : c.value : x(c) ? e ? Ut(c) : Ht(c) : c;
  };
}
const Lr = /* @__PURE__ */ Yr();
function Yr(e = !1) {
  return function(r, a, o, n) {
    let l = r[a];
    if (q(l) && E(l) && !E(o))
      return !1;
    if (!e && (!Re(o) && !q(o) && (l = f(l), o = f(o)), !b(r) && E(l) && !E(o)))
      return l.value = o, !0;
    const c = b(r) && Ke(a) ? Number(a) < r.length : p(r, a), u = Reflect.set(r, a, o, n);
    return r === f(n) && (c ? ke(o, l) && A(r, "set", a, o, l) : A(r, "add", a, o)), u;
  };
}
function Qr(e, t) {
  const r = p(e, t), a = e[t], o = Reflect.deleteProperty(e, t);
  return o && r && A(e, "delete", t, void 0, a), o;
}
function Xr(e, t) {
  const r = Reflect.has(e, t);
  return (!je(t) || !Mt.has(t)) && N(e, "has", t), r;
}
function Zr(e) {
  return N(e, "iterate", b(e) ? "length" : H), Reflect.ownKeys(e);
}
const eo = {
  get: Wr,
  set: Lr,
  deleteProperty: Qr,
  has: Xr,
  ownKeys: Zr
}, zt = {
  get: Br,
  set(e, t) {
    return process.env.NODE_ENV !== "production" && ot(
      `Set operation on key "${String(t)}" failed: target is readonly.`,
      e
    ), !0;
  },
  deleteProperty(e, t) {
    return process.env.NODE_ENV !== "production" && ot(
      `Delete operation on key "${String(t)}" failed: target is readonly.`,
      e
    ), !0;
  }
}, to = /* @__PURE__ */ C(
  {},
  zt,
  {
    get: Jr
  }
), Be = (e) => e, Ne = (e) => Reflect.getPrototypeOf(e);
function ce(e, t, r = !1, a = !1) {
  e = e.__v_raw;
  const o = f(e), n = f(t);
  r || (t !== n && N(o, "get", t), N(o, "get", n));
  const { has: l } = Ne(o), c = a ? Be : r ? Le : Ge;
  if (l.call(o, t))
    return c(e.get(t));
  if (l.call(o, n))
    return c(e.get(n));
  e !== o && e.get(t);
}
function de(e, t = !1) {
  const r = this.__v_raw, a = f(r), o = f(e);
  return t || (e !== o && N(a, "has", e), N(a, "has", o)), e === o ? r.has(e) : r.has(e) || r.has(o);
}
function ue(e, t = !1) {
  return e = e.__v_raw, !t && N(f(e), "iterate", H), Reflect.get(e, "size", e);
}
function ct(e) {
  e = f(e);
  const t = f(this);
  return Ne(t).has.call(t, e) || (t.add(e), A(t, "add", e, e)), this;
}
function dt(e, t) {
  t = f(t);
  const r = f(this), { has: a, get: o } = Ne(r);
  let n = a.call(r, e);
  n ? process.env.NODE_ENV !== "production" && Ft(r, a, e) : (e = f(e), n = a.call(r, e));
  const l = o.call(r, e);
  return r.set(e, t), n ? ke(t, l) && A(r, "set", e, t, l) : A(r, "add", e, t), this;
}
function ut(e) {
  const t = f(this), { has: r, get: a } = Ne(t);
  let o = r.call(t, e);
  o ? process.env.NODE_ENV !== "production" && Ft(t, r, e) : (e = f(e), o = r.call(t, e));
  const n = a ? a.call(t, e) : void 0, l = t.delete(e);
  return o && A(t, "delete", e, void 0, n), l;
}
function it() {
  const e = f(this), t = e.size !== 0, r = process.env.NODE_ENV !== "production" ? K(e) ? new Map(e) : new Set(e) : void 0, a = e.clear();
  return t && A(e, "clear", void 0, void 0, r), a;
}
function ie(e, t) {
  return function(a, o) {
    const n = this, l = n.__v_raw, c = f(l), u = t ? Be : e ? Le : Ge;
    return !e && N(c, "iterate", H), l.forEach((i, h) => a.call(o, u(i), u(h), n));
  };
}
function se(e, t, r) {
  return function(...a) {
    const o = this.__v_raw, n = f(o), l = K(n), c = e === "entries" || e === Symbol.iterator && l, u = e === "keys" && l, i = o[e](...a), h = r ? Be : t ? Le : Ge;
    return !t && N(
      n,
      "iterate",
      u ? Ie : H
    ), {
      // iterator protocol
      next() {
        const { value: d, done: s } = i.next();
        return s ? { value: d, done: s } : {
          value: c ? [h(d[0]), h(d[1])] : h(d),
          done: s
        };
      },
      // iterable protocol
      [Symbol.iterator]() {
        return this;
      }
    };
  };
}
function P(e) {
  return function(...t) {
    if (process.env.NODE_ENV !== "production") {
      const r = t[0] ? `on key "${t[0]}" ` : "";
      console.warn(
        `${pe(e)} operation ${r}failed: target is readonly.`,
        f(this)
      );
    }
    return e === "delete" ? !1 : this;
  };
}
function ro() {
  const e = {
    get(n) {
      return ce(this, n);
    },
    get size() {
      return ue(this);
    },
    has: de,
    add: ct,
    set: dt,
    delete: ut,
    clear: it,
    forEach: ie(!1, !1)
  }, t = {
    get(n) {
      return ce(this, n, !1, !0);
    },
    get size() {
      return ue(this);
    },
    has: de,
    add: ct,
    set: dt,
    delete: ut,
    clear: it,
    forEach: ie(!1, !0)
  }, r = {
    get(n) {
      return ce(this, n, !0);
    },
    get size() {
      return ue(this, !0);
    },
    has(n) {
      return de.call(this, n, !0);
    },
    add: P("add"),
    set: P("set"),
    delete: P("delete"),
    clear: P("clear"),
    forEach: ie(!0, !1)
  }, a = {
    get(n) {
      return ce(this, n, !0, !0);
    },
    get size() {
      return ue(this, !0);
    },
    has(n) {
      return de.call(this, n, !0);
    },
    add: P("add"),
    set: P("set"),
    delete: P("delete"),
    clear: P("clear"),
    forEach: ie(!0, !0)
  };
  return ["keys", "values", "entries", Symbol.iterator].forEach((n) => {
    e[n] = se(
      n,
      !1,
      !1
    ), r[n] = se(
      n,
      !0,
      !1
    ), t[n] = se(
      n,
      !1,
      !0
    ), a[n] = se(
      n,
      !0,
      !0
    );
  }), [
    e,
    r,
    t,
    a
  ];
}
const [
  oo,
  ao,
  no,
  lo
] = /* @__PURE__ */ ro();
function Je(e, t) {
  const r = t ? e ? lo : no : e ? ao : oo;
  return (a, o, n) => o === "__v_isReactive" ? !e : o === "__v_isReadonly" ? e : o === "__v_raw" ? a : Reflect.get(
    p(r, o) && o in a ? r : a,
    o,
    n
  );
}
const co = {
  get: /* @__PURE__ */ Je(!1, !1)
}, uo = {
  get: /* @__PURE__ */ Je(!0, !1)
}, io = {
  get: /* @__PURE__ */ Je(!0, !0)
};
function Ft(e, t, r) {
  const a = f(r);
  if (a !== r && t.call(e, a)) {
    const o = Ot(e);
    console.warn(
      `Reactive ${o} contains both the raw and reactive versions of the same object${o === "Map" ? " as keys" : ""}, which can lead to inconsistencies. Avoid differentiating between the raw and reactive versions of an object and only use the reactive version if possible.`
    );
  }
}
const At = /* @__PURE__ */ new WeakMap(), so = /* @__PURE__ */ new WeakMap(), jt = /* @__PURE__ */ new WeakMap(), Kt = /* @__PURE__ */ new WeakMap();
function fo(e) {
  switch (e) {
    case "Object":
    case "Array":
      return 1;
    case "Map":
    case "Set":
    case "WeakMap":
    case "WeakSet":
      return 2;
    default:
      return 0;
  }
}
function ho(e) {
  return e.__v_skip || !Object.isExtensible(e) ? 0 : fo(Ot(e));
}
function Ht(e) {
  return q(e) ? e : qe(
    e,
    !1,
    eo,
    co,
    At
  );
}
function Ut(e) {
  return qe(
    e,
    !0,
    zt,
    uo,
    jt
  );
}
function fe(e) {
  return qe(
    e,
    !0,
    to,
    io,
    Kt
  );
}
function qe(e, t, r, a, o) {
  if (!x(e))
    return process.env.NODE_ENV !== "production" && console.warn(`value cannot be made reactive: ${String(e)}`), e;
  if (e.__v_raw && !(t && e.__v_isReactive))
    return e;
  const n = o.get(e);
  if (n)
    return n;
  const l = ho(e);
  if (l === 0)
    return e;
  const c = new Proxy(
    e,
    l === 2 ? a : r
  );
  return o.set(e, c), c;
}
function W(e) {
  return q(e) ? W(e.__v_raw) : !!(e && e.__v_isReactive);
}
function q(e) {
  return !!(e && e.__v_isReadonly);
}
function Re(e) {
  return !!(e && e.__v_isShallow);
}
function $e(e) {
  return W(e) || q(e);
}
function f(e) {
  const t = e && e.__v_raw;
  return t ? f(t) : e;
}
function bo(e) {
  return Dr(e, "__v_skip", !0), e;
}
const Ge = (e) => x(e) ? Ht(e) : e, Le = (e) => x(e) ? Ut(e) : e;
function E(e) {
  return !!(e && e.__v_isRef === !0);
}
function Wt(e) {
  return E(e) ? e.value : e;
}
const mo = {
  get: (e, t, r) => Wt(Reflect.get(e, t, r)),
  set: (e, t, r, a) => {
    const o = e[t];
    return E(o) && !E(r) ? (o.value = r, !0) : Reflect.set(e, t, r, a);
  }
};
function go(e) {
  return W(e) ? e : new Proxy(e, mo);
}
const B = [];
function po(e) {
  B.push(e);
}
function ko() {
  B.pop();
}
function k(e, ...t) {
  if (process.env.NODE_ENV === "production")
    return;
  $t();
  const r = B.length ? B[B.length - 1].component : null, a = r && r.appContext.config.warnHandler, o = vo();
  if (a)
    J(
      a,
      r,
      11,
      [
        e + t.join(""),
        r && r.proxy,
        o.map(
          ({ vnode: n }) => `at <${ur(r, n.type)}>`
        ).join(`
`),
        o
      ]
    );
  else {
    const n = [`[Vue warn]: ${e}`, ...t];
    o.length && n.push(`
`, ...xo(o)), console.warn(...n);
  }
  Pt();
}
function vo() {
  let e = B[B.length - 1];
  if (!e)
    return [];
  const t = [];
  for (; e; ) {
    const r = t[0];
    r && r.vnode === e ? r.recurseCount++ : t.push({
      vnode: e,
      recurseCount: 0
    });
    const a = e.component && e.component.parent;
    e = a && a.vnode;
  }
  return t;
}
function xo(e) {
  const t = [];
  return e.forEach((r, a) => {
    t.push(...a === 0 ? [] : [`
`], ..._o(r));
  }), t;
}
function _o({ vnode: e, recurseCount: t }) {
  const r = t > 0 ? `... (${t} recursive calls)` : "", a = e.component ? e.component.parent == null : !1, o = ` at <${ur(
    e.component,
    e.type,
    a
  )}`, n = ">" + r;
  return e.props ? [o, ...wo(e.props), n] : [o + n];
}
function wo(e) {
  const t = [], r = Object.keys(e);
  return r.slice(0, 3).forEach((a) => {
    t.push(...Bt(a, e[a]));
  }), r.length > 3 && t.push(" ..."), t;
}
function Bt(e, t, r) {
  return S(t) ? (t = JSON.stringify(t), r ? t : [`${e}=${t}`]) : typeof t == "number" || typeof t == "boolean" || t == null ? r ? t : [`${e}=${t}`] : E(t) ? (t = Bt(e, f(t.value), !0), r ? t : [`${e}=Ref<`, t, ">"]) : v(t) ? [`${e}=fn${t.name ? `<${t.name}>` : ""}`] : (t = f(t), r ? t : [`${e}=`, t]);
}
const Jt = {
  sp: "serverPrefetch hook",
  bc: "beforeCreate hook",
  c: "created hook",
  bm: "beforeMount hook",
  m: "mounted hook",
  bu: "beforeUpdate hook",
  u: "updated",
  bum: "beforeUnmount hook",
  um: "unmounted hook",
  a: "activated hook",
  da: "deactivated hook",
  ec: "errorCaptured hook",
  rtc: "renderTracked hook",
  rtg: "renderTriggered hook",
  0: "setup function",
  1: "render function",
  2: "watcher getter",
  3: "watcher callback",
  4: "watcher cleanup function",
  5: "native event handler",
  6: "component event handler",
  7: "vnode hook",
  8: "directive hook",
  9: "transition hook",
  10: "app errorHandler",
  11: "app warnHandler",
  12: "ref function",
  13: "async component loader",
  14: "scheduler flush. This is likely a Vue internals bug. Please open an issue at https://new-issue.vuejs.org/?repo=vuejs/core"
};
function J(e, t, r, a) {
  let o;
  try {
    o = a ? e(...a) : e();
  } catch (n) {
    qt(n, t, r);
  }
  return o;
}
function Pe(e, t, r, a) {
  if (v(e)) {
    const n = J(e, t, r, a);
    return n && Vr(n) && n.catch((l) => {
      qt(l, t, r);
    }), n;
  }
  const o = [];
  for (let n = 0; n < e.length; n++)
    o.push(Pe(e[n], t, r, a));
  return o;
}
function qt(e, t, r, a = !0) {
  const o = t ? t.vnode : null;
  if (t) {
    let n = t.parent;
    const l = t.proxy, c = process.env.NODE_ENV !== "production" ? Jt[r] : r;
    for (; n; ) {
      const i = n.ec;
      if (i) {
        for (let h = 0; h < i.length; h++)
          if (i[h](e, l, c) === !1)
            return;
      }
      n = n.parent;
    }
    const u = t.appContext.config.errorHandler;
    if (u) {
      J(
        u,
        null,
        10,
        [e, l, c]
      );
      return;
    }
  }
  Eo(e, r, o, a);
}
function Eo(e, t, r, a = !0) {
  if (process.env.NODE_ENV !== "production") {
    const o = Jt[t];
    if (r && po(r), k(`Unhandled error${o ? ` during execution of ${o}` : ""}`), r && ko(), a)
      throw e;
    console.error(e);
  } else
    console.error(e);
}
let ve = !1, Me = !1;
const D = [];
let z = 0;
const Q = [];
let R = null, M = 0;
const Gt = /* @__PURE__ */ Promise.resolve();
let Ye = null;
const No = 100;
function yo(e) {
  const t = Ye || Gt;
  return e ? t.then(this ? e.bind(this) : e) : t;
}
function Oo(e) {
  let t = z + 1, r = D.length;
  for (; t < r; ) {
    const a = t + r >>> 1;
    oe(D[a]) < e ? t = a + 1 : r = a;
  }
  return t;
}
function Qe(e) {
  (!D.length || !D.includes(
    e,
    ve && e.allowRecurse ? z + 1 : z
  )) && (e.id == null ? D.push(e) : D.splice(Oo(e.id), 0, e), Lt());
}
function Lt() {
  !ve && !Me && (Me = !0, Ye = Gt.then(Qt));
}
function Yt(e) {
  b(e) ? Q.push(...e) : (!R || !R.includes(
    e,
    e.allowRecurse ? M + 1 : M
  )) && Q.push(e), Lt();
}
function So(e) {
  if (Q.length) {
    const t = [...new Set(Q)];
    if (Q.length = 0, R) {
      R.push(...t);
      return;
    }
    for (R = t, process.env.NODE_ENV !== "production" && (e = e || /* @__PURE__ */ new Map()), R.sort((r, a) => oe(r) - oe(a)), M = 0; M < R.length; M++)
      process.env.NODE_ENV !== "production" && Xt(e, R[M]) || R[M]();
    R = null, M = 0;
  }
}
const oe = (e) => e.id == null ? 1 / 0 : e.id, Co = (e, t) => {
  const r = oe(e) - oe(t);
  if (r === 0) {
    if (e.pre && !t.pre)
      return -1;
    if (t.pre && !e.pre)
      return 1;
  }
  return r;
};
function Qt(e) {
  Me = !1, ve = !0, process.env.NODE_ENV !== "production" && (e = e || /* @__PURE__ */ new Map()), D.sort(Co);
  const t = process.env.NODE_ENV !== "production" ? (r) => Xt(e, r) : Et;
  try {
    for (z = 0; z < D.length; z++) {
      const r = D[z];
      if (r && r.active !== !1) {
        if (process.env.NODE_ENV !== "production" && t(r))
          continue;
        J(r, null, 14);
      }
    }
  } finally {
    z = 0, D.length = 0, So(e), ve = !1, Ye = null, (D.length || Q.length) && Qt(e);
  }
}
function Xt(e, t) {
  if (!e.has(t))
    e.set(t, 1);
  else {
    const r = e.get(t);
    if (r > No) {
      const a = t.ownerInstance, o = a && tt(a.type);
      return k(
        `Maximum recursive updates exceeded${o ? ` in component <${o}>` : ""}. This means you have a reactive effect that is mutating its own dependencies and thus recursively triggering itself. Possible sources include component template, render function, updated hook or watcher source function.`
      ), !0;
    } else
      e.set(t, r + 1);
  }
}
const X = /* @__PURE__ */ new Set();
process.env.NODE_ENV !== "production" && (Ce().__VUE_HMR_RUNTIME__ = {
  createRecord: ye(Vo),
  rerender: ye(To),
  reload: ye(Do)
});
const xe = /* @__PURE__ */ new Map();
function Vo(e, t) {
  return xe.has(e) ? !1 : (xe.set(e, {
    initialDef: te(t),
    instances: /* @__PURE__ */ new Set()
  }), !0);
}
function te(e) {
  return ir(e) ? e.__vccOpts : e;
}
function To(e, t) {
  const r = xe.get(e);
  r && (r.initialDef.render = t, [...r.instances].forEach((a) => {
    t && (a.render = t, te(a.type).render = t), a.renderCache = [], a.update();
  }));
}
function Do(e, t) {
  const r = xe.get(e);
  if (!r)
    return;
  t = te(t), st(r.initialDef, t);
  const a = [...r.instances];
  for (const o of a) {
    const n = te(o.type);
    X.has(n) || (n !== r.initialDef && st(n, t), X.add(n)), o.appContext.propsCache.delete(o.type), o.appContext.emitsCache.delete(o.type), o.appContext.optionsCache.delete(o.type), o.ceReload ? (X.add(n), o.ceReload(t.styles), X.delete(n)) : o.parent ? Qe(o.parent.update) : o.appContext.reload ? o.appContext.reload() : typeof window < "u" ? window.location.reload() : console.warn(
      "[HMR] Root or manually mounted instance modified. Full reload required."
    );
  }
  Yt(() => {
    for (const o of a)
      X.delete(
        te(o.type)
      );
  });
}
function st(e, t) {
  C(e, t);
  for (const r in e)
    r !== "__file" && !(r in t) && delete e[r];
}
function ye(e) {
  return (t, r) => {
    try {
      return e(t, r);
    } catch (a) {
      console.error(a), console.warn(
        "[HMR] Something went wrong during Vue component hot-reload. Full reload required."
      );
    }
  };
}
function Io(e, ...t) {
}
const Ro = /* @__PURE__ */ $o(
  "component:updated"
  /* COMPONENT_UPDATED */
);
function $o(e) {
  return (t) => {
    Io(
      e,
      t.appContext.app,
      t.uid,
      t.parent ? t.parent.uid : void 0,
      t
    );
  };
}
let T = null, Zt = null;
function ft(e) {
  const t = T;
  return T = e, Zt = e && e.type.__scopeId || null, t;
}
function Po(e, t = T, r) {
  if (!t || e._n)
    return e;
  const a = (...o) => {
    a._d && vt(-1);
    const n = ft(t);
    let l;
    try {
      l = e(...o);
    } finally {
      ft(n), a._d && vt(1);
    }
    return process.env.NODE_ENV !== "production" && Ro(t), l;
  };
  return a._n = !0, a._c = !0, a._d = !0, a;
}
const Mo = (e) => e.__isSuspense;
function zo(e, t) {
  t && t.pendingBranch ? b(e) ? t.effects.push(...e) : t.effects.push(e) : Yt(e);
}
const he = {};
function Fo(e, t, { immediate: r, deep: a, flush: o, onTrack: n, onTrigger: l } = $) {
  var c;
  process.env.NODE_ENV !== "production" && !t && (r !== void 0 && k(
    'watch() "immediate" option is only respected when using the watch(source, callback, options?) signature.'
  ), a !== void 0 && k(
    'watch() "deep" option is only respected when using the watch(source, callback, options?) signature.'
  ));
  const u = (g) => {
    k(
      "Invalid watch source: ",
      g,
      "A watch source can only be a getter/effect function, a ref, a reactive object, or an array of these types."
    );
  }, i = Fr() === ((c = G) == null ? void 0 : c.scope) ? G : null;
  let h, d = !1, s = !1;
  if (E(e) ? (h = () => e.value, d = Re(e)) : W(e) ? (h = () => e, a = !0) : b(e) ? (s = !0, d = e.some((g) => W(g) || Re(g)), h = () => e.map((g) => {
    if (E(g))
      return g.value;
    if (W(g))
      return Y(g);
    if (v(g))
      return J(g, i, 2);
    process.env.NODE_ENV !== "production" && u(g);
  })) : v(e) ? t ? h = () => J(e, i, 2) : h = () => {
    if (!(i && i.isUnmounted))
      return m && m(), Pe(
        e,
        i,
        3,
        [y]
      );
  } : (h = Et, process.env.NODE_ENV !== "production" && u(e)), t && a) {
    const g = h;
    h = () => Y(g());
  }
  let m, y = (g) => {
    m = V.onStop = () => {
      J(g, i, 4);
    };
  }, O = s ? new Array(e.length).fill(he) : he;
  const j = () => {
    if (V.active)
      if (t) {
        const g = V.run();
        (a || d || (s ? g.some(
          (fr, hr) => ke(fr, O[hr])
        ) : ke(g, O))) && (m && m(), Pe(t, i, 3, [
          g,
          // pass undefined as the old value when it's changed for the first time
          O === he ? void 0 : s && O[0] === he ? [] : O,
          y
        ]), O = g);
      } else
        V.run();
  };
  j.allowRecurse = !!t;
  let ne;
  o === "sync" ? ne = j : o === "post" ? ne = () => kt(j, i && i.suspense) : (j.pre = !0, i && (j.id = i.uid), ne = () => Qe(j));
  const V = new Kr(h, ne);
  return process.env.NODE_ENV !== "production" && (V.onTrack = n, V.onTrigger = l), t ? r ? j() : O = V.run() : o === "post" ? kt(
    V.run.bind(V),
    i && i.suspense
  ) : V.run(), () => {
    V.stop(), i && i.scope && Sr(i.scope.effects, V);
  };
}
function Ao(e, t, r) {
  const a = this.proxy, o = S(e) ? e.includes(".") ? jo(a, e) : () => a[e] : e.bind(a, a);
  let n;
  v(t) ? n = t : (n = t.handler, r = t);
  const l = G;
  _t(this);
  const c = Fo(o, n.bind(a), r);
  return l ? _t(l) : la(), c;
}
function jo(e, t) {
  const r = t.split(".");
  return () => {
    let a = e;
    for (let o = 0; o < r.length && a; o++)
      a = a[r[o]];
    return a;
  };
}
function Y(e, t) {
  if (!x(e) || e.__v_skip || (t = t || /* @__PURE__ */ new Set(), t.has(e)))
    return e;
  if (t.add(e), E(e))
    Y(e.value, t);
  else if (b(e))
    for (let r = 0; r < e.length; r++)
      Y(e[r], t);
  else if (Nt(e) || K(e))
    e.forEach((r) => {
      Y(r, t);
    });
  else if (St(e))
    for (const r in e)
      Y(e[r], t);
  return e;
}
function Ko(e, t) {
  return v(e) ? (
    // #8326: extend call and options.name access are considered side-effects
    // by Rollup, so we have to wrap it in a pure-annotated IIFE.
    /* @__PURE__ */ (() => C({ name: e.name }, t, { setup: e }))()
  ) : e;
}
const ze = "components", er = Symbol.for("v-ndc");
function Ho(e) {
  return S(e) ? Uo(ze, e, !1) || e : e || er;
}
function Uo(e, t, r = !0, a = !1) {
  const o = T || G;
  if (o) {
    const n = o.type;
    if (e === ze) {
      const c = tt(
        n,
        !1
        /* do not include inferred name to avoid breaking existing code */
      );
      if (c && (c === t || c === ge(t) || c === pe(ge(t))))
        return n;
    }
    const l = (
      // local registration
      // check instance[type] first which is resolved for options API
      ht(o[e] || n[e], t) || // global registration
      ht(o.appContext[e], t)
    );
    if (!l && a)
      return n;
    if (process.env.NODE_ENV !== "production" && r && !l) {
      const c = e === ze ? `
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.` : "";
      k(`Failed to resolve ${e.slice(0, -1)}: ${t}${c}`);
    }
    return l;
  } else
    process.env.NODE_ENV !== "production" && k(
      `resolve${pe(e.slice(0, -1))} can only be used in render() or setup().`
    );
}
function ht(e, t) {
  return e && (e[t] || e[ge(t)] || e[pe(ge(t))]);
}
const Fe = (e) => e ? ca(e) ? da(e) || e.proxy : Fe(e.parent) : null, re = (
  // Move PURE marker to new line to workaround compiler discarding it
  // due to type annotation
  /* @__PURE__ */ C(/* @__PURE__ */ Object.create(null), {
    $: (e) => e,
    $el: (e) => e.vnode.el,
    $data: (e) => e.data,
    $props: (e) => process.env.NODE_ENV !== "production" ? fe(e.props) : e.props,
    $attrs: (e) => process.env.NODE_ENV !== "production" ? fe(e.attrs) : e.attrs,
    $slots: (e) => process.env.NODE_ENV !== "production" ? fe(e.slots) : e.slots,
    $refs: (e) => process.env.NODE_ENV !== "production" ? fe(e.refs) : e.refs,
    $parent: (e) => Fe(e.parent),
    $root: (e) => Fe(e.root),
    $emit: (e) => e.emit,
    $options: (e) => Jo(e),
    $forceUpdate: (e) => e.f || (e.f = () => Qe(e.update)),
    $nextTick: (e) => e.n || (e.n = yo.bind(e.proxy)),
    $watch: (e) => Ao.bind(e)
  })
), Wo = (e) => e === "_" || e === "$", Oe = (e, t) => e !== $ && !e.__isScriptSetup && p(e, t), Bo = {
  get({ _: e }, t) {
    const { ctx: r, setupState: a, data: o, props: n, accessCache: l, type: c, appContext: u } = e;
    if (process.env.NODE_ENV !== "production" && t === "__isVue")
      return !0;
    let i;
    if (t[0] !== "$") {
      const m = l[t];
      if (m !== void 0)
        switch (m) {
          case 1:
            return a[t];
          case 2:
            return o[t];
          case 4:
            return r[t];
          case 3:
            return n[t];
        }
      else {
        if (Oe(a, t))
          return l[t] = 1, a[t];
        if (o !== $ && p(o, t))
          return l[t] = 2, o[t];
        if (
          // only cache other properties when instance has declared (thus stable)
          // props
          (i = e.propsOptions[0]) && p(i, t)
        )
          return l[t] = 3, n[t];
        if (r !== $ && p(r, t))
          return l[t] = 4, r[t];
        l[t] = 0;
      }
    }
    const h = re[t];
    let d, s;
    if (h)
      return t === "$attrs" ? (N(e, "get", t), process.env.NODE_ENV !== "production" && void 0) : process.env.NODE_ENV !== "production" && t === "$slots" && N(e, "get", t), h(e);
    if (
      // css module (injected by vue-loader)
      (d = c.__cssModules) && (d = d[t])
    )
      return d;
    if (r !== $ && p(r, t))
      return l[t] = 4, r[t];
    if (
      // global properties
      s = u.config.globalProperties, p(s, t)
    )
      return s[t];
    process.env.NODE_ENV !== "production" && T && (!S(t) || // #1091 avoid internal isRef/isVNode checks on component instance leading
    // to infinite warning loop
    t.indexOf("__v") !== 0) && (o !== $ && Wo(t[0]) && p(o, t) ? k(
      `Property ${JSON.stringify(
        t
      )} must be accessed via $data because it starts with a reserved character ("$" or "_") and is not proxied on the render context.`
    ) : e === T && k(
      `Property ${JSON.stringify(t)} was accessed during render but is not defined on instance.`
    ));
  },
  set({ _: e }, t, r) {
    const { data: a, setupState: o, ctx: n } = e;
    return Oe(o, t) ? (o[t] = r, !0) : process.env.NODE_ENV !== "production" && o.__isScriptSetup && p(o, t) ? (k(`Cannot mutate <script setup> binding "${t}" from Options API.`), !1) : a !== $ && p(a, t) ? (a[t] = r, !0) : p(e.props, t) ? (process.env.NODE_ENV !== "production" && k(`Attempting to mutate prop "${t}". Props are readonly.`), !1) : t[0] === "$" && t.slice(1) in e ? (process.env.NODE_ENV !== "production" && k(
      `Attempting to mutate public property "${t}". Properties starting with $ are reserved and readonly.`
    ), !1) : (process.env.NODE_ENV !== "production" && t in e.appContext.config.globalProperties ? Object.defineProperty(n, t, {
      enumerable: !0,
      configurable: !0,
      value: r
    }) : n[t] = r, !0);
  },
  has({
    _: { data: e, setupState: t, accessCache: r, ctx: a, appContext: o, propsOptions: n }
  }, l) {
    let c;
    return !!r[l] || e !== $ && p(e, l) || Oe(t, l) || (c = n[0]) && p(c, l) || p(a, l) || p(re, l) || p(o.config.globalProperties, l);
  },
  defineProperty(e, t, r) {
    return r.get != null ? e._.accessCache[t] = 0 : p(r, "value") && this.set(e, t, r.value, null), Reflect.defineProperty(e, t, r);
  }
};
process.env.NODE_ENV !== "production" && (Bo.ownKeys = (e) => (k(
  "Avoid app logic that relies on enumerating keys on a component instance. The keys will be empty in production mode to avoid performance overhead."
), Reflect.ownKeys(e)));
function bt(e) {
  return b(e) ? e.reduce(
    (t, r) => (t[r] = null, t),
    {}
  ) : e;
}
function Jo(e) {
  const t = e.type, { mixins: r, extends: a } = t, {
    mixins: o,
    optionsCache: n,
    config: { optionMergeStrategies: l }
  } = e.appContext, c = n.get(t);
  let u;
  return c ? u = c : !o.length && !r && !a ? u = t : (u = {}, o.length && o.forEach(
    (i) => _e(u, i, l, !0)
  ), _e(u, t, l)), x(t) && n.set(t, u), u;
}
function _e(e, t, r, a = !1) {
  const { mixins: o, extends: n } = t;
  n && _e(e, n, r, !0), o && o.forEach(
    (l) => _e(e, l, r, !0)
  );
  for (const l in t)
    if (a && l === "expose")
      process.env.NODE_ENV !== "production" && k(
        '"expose" option is ignored when declared in mixins or extends. It should only be declared in the base component itself.'
      );
    else {
      const c = qo[l] || r && r[l];
      e[l] = c ? c(e[l], t[l]) : t[l];
    }
  return e;
}
const qo = {
  data: mt,
  props: pt,
  emits: pt,
  // objects
  methods: ee,
  computed: ee,
  // lifecycle
  beforeCreate: _,
  created: _,
  beforeMount: _,
  mounted: _,
  beforeUpdate: _,
  updated: _,
  beforeDestroy: _,
  beforeUnmount: _,
  destroyed: _,
  unmounted: _,
  activated: _,
  deactivated: _,
  errorCaptured: _,
  serverPrefetch: _,
  // assets
  components: ee,
  directives: ee,
  // watch
  watch: Lo,
  // provide / inject
  provide: mt,
  inject: Go
};
function mt(e, t) {
  return t ? e ? function() {
    return C(
      v(e) ? e.call(this, this) : e,
      v(t) ? t.call(this, this) : t
    );
  } : t : e;
}
function Go(e, t) {
  return ee(gt(e), gt(t));
}
function gt(e) {
  if (b(e)) {
    const t = {};
    for (let r = 0; r < e.length; r++)
      t[e[r]] = e[r];
    return t;
  }
  return e;
}
function _(e, t) {
  return e ? [...new Set([].concat(e, t))] : t;
}
function ee(e, t) {
  return e ? C(/* @__PURE__ */ Object.create(null), e, t) : t;
}
function pt(e, t) {
  return e ? b(e) && b(t) ? [.../* @__PURE__ */ new Set([...e, ...t])] : C(
    /* @__PURE__ */ Object.create(null),
    bt(e),
    bt(t ?? {})
  ) : t;
}
function Lo(e, t) {
  if (!e)
    return t;
  if (!t)
    return e;
  const r = C(/* @__PURE__ */ Object.create(null), e);
  for (const a in t)
    r[a] = _(e[a], t[a]);
  return r;
}
const kt = zo, Yo = (e) => e.__isTeleport, tr = Symbol.for("v-fgt"), Qo = Symbol.for("v-txt"), Ae = Symbol.for("v-cmt"), be = [];
let I = null;
function rr(e = !1) {
  be.push(I = e ? null : []);
}
function Xo() {
  be.pop(), I = be[be.length - 1] || null;
}
let ae = 1;
function vt(e) {
  ae += e;
}
function Zo(e) {
  return e.dynamicChildren = ae > 0 ? I || Nr : null, Xo(), ae > 0 && I && I.push(e), e;
}
function or(e, t, r, a, o) {
  return Zo(
    Xe(
      e,
      t,
      r,
      a,
      o,
      !0
      /* isBlock: prevent a block from tracking itself */
    )
  );
}
function ea(e) {
  return e ? e.__v_isVNode === !0 : !1;
}
const ta = (...e) => lr(
  ...e
), ar = "__vInternal", nr = ({ key: e }) => e ?? null, me = ({
  ref: e,
  ref_key: t,
  ref_for: r
}) => (typeof e == "number" && (e = "" + e), e != null ? S(e) || E(e) || v(e) ? { i: T, r: e, k: t, f: !!r } : e : null);
function ra(e, t = null, r = null, a = 0, o = null, n = e === tr ? 0 : 1, l = !1, c = !1) {
  const u = {
    __v_isVNode: !0,
    __v_skip: !0,
    type: e,
    props: t,
    key: t && nr(t),
    ref: t && me(t),
    scopeId: Zt,
    slotScopeIds: null,
    children: r,
    component: null,
    suspense: null,
    ssContent: null,
    ssFallback: null,
    dirs: null,
    transition: null,
    el: null,
    anchor: null,
    target: null,
    targetAnchor: null,
    staticCount: 0,
    shapeFlag: n,
    patchFlag: a,
    dynamicProps: o,
    dynamicChildren: null,
    appContext: null,
    ctx: T
  };
  return c ? (Ze(u, r), n & 128 && e.normalize(u)) : r && (u.shapeFlag |= S(r) ? 8 : 16), process.env.NODE_ENV !== "production" && u.key !== u.key && k("VNode created with invalid key (NaN). VNode type:", u.type), ae > 0 && // avoid a block node from tracking itself
  !l && // has current parent block
  I && // presence of a patch flag indicates this node needs patching on updates.
  // component nodes also should always be patched, because even if the
  // component doesn't need to update, it needs to persist the instance on to
  // the next vnode so that it can be properly unmounted later.
  (u.patchFlag > 0 || n & 6) && // the EVENTS flag is only for hydration and if it is the only flag, the
  // vnode should not be considered dynamic due to handler caching.
  u.patchFlag !== 32 && I.push(u), u;
}
const Xe = process.env.NODE_ENV !== "production" ? ta : lr;
function lr(e, t = null, r = null, a = 0, o = null, n = !1) {
  if ((!e || e === er) && (process.env.NODE_ENV !== "production" && !e && k(`Invalid vnode type when creating vnode: ${e}.`), e = Ae), ea(e)) {
    const c = we(
      e,
      t,
      !0
      /* mergeRef: true */
    );
    return r && Ze(c, r), ae > 0 && !n && I && (c.shapeFlag & 6 ? I[I.indexOf(e)] = c : I.push(c)), c.patchFlag |= -2, c;
  }
  if (ir(e) && (e = e.__vccOpts), t) {
    t = oa(t);
    let { class: c, style: u } = t;
    c && !S(c) && (t.class = Ue(c)), x(u) && ($e(u) && !b(u) && (u = C({}, u)), t.style = He(u));
  }
  const l = S(e) ? 1 : Mo(e) ? 128 : Yo(e) ? 64 : x(e) ? 4 : v(e) ? 2 : 0;
  return process.env.NODE_ENV !== "production" && l & 4 && $e(e) && (e = f(e), k(
    "Vue received a Component which was made a reactive object. This can lead to unnecessary performance overhead, and should be avoided by marking the component with `markRaw` or using `shallowRef` instead of `ref`.",
    `
Component that was made reactive: `,
    e
  )), ra(
    e,
    t,
    r,
    a,
    o,
    l,
    n,
    !0
  );
}
function oa(e) {
  return e ? $e(e) || ar in e ? C({}, e) : e : null;
}
function we(e, t, r = !1) {
  const { props: a, ref: o, patchFlag: n, children: l } = e, c = t ? na(a || {}, t) : a;
  return {
    __v_isVNode: !0,
    __v_skip: !0,
    type: e.type,
    props: c,
    key: c && nr(c),
    ref: t && t.ref ? (
      // #2078 in the case of <component :is="vnode" ref="extra"/>
      // if the vnode itself already has a ref, cloneVNode will need to merge
      // the refs so the single vnode can be set on multiple refs
      r && o ? b(o) ? o.concat(me(t)) : [o, me(t)] : me(t)
    ) : o,
    scopeId: e.scopeId,
    slotScopeIds: e.slotScopeIds,
    children: process.env.NODE_ENV !== "production" && n === -1 && b(l) ? l.map(cr) : l,
    target: e.target,
    targetAnchor: e.targetAnchor,
    staticCount: e.staticCount,
    shapeFlag: e.shapeFlag,
    // if the vnode is cloned with extra props, we can no longer assume its
    // existing patch flag to be reliable and need to add the FULL_PROPS flag.
    // note: preserve flag for fragments since they use the flag for children
    // fast paths only.
    patchFlag: t && e.type !== tr ? n === -1 ? 16 : n | 16 : n,
    dynamicProps: e.dynamicProps,
    dynamicChildren: e.dynamicChildren,
    appContext: e.appContext,
    dirs: e.dirs,
    transition: e.transition,
    // These should technically only be non-null on mounted VNodes. However,
    // they *should* be copied for kept-alive vnodes. So we just always copy
    // them since them being non-null during a mount doesn't affect the logic as
    // they will simply be overwritten.
    component: e.component,
    suspense: e.suspense,
    ssContent: e.ssContent && we(e.ssContent),
    ssFallback: e.ssFallback && we(e.ssFallback),
    el: e.el,
    anchor: e.anchor,
    ctx: e.ctx,
    ce: e.ce
  };
}
function cr(e) {
  const t = we(e);
  return b(e.children) && (t.children = e.children.map(cr)), t;
}
function dr(e = " ", t = 0) {
  return Xe(Qo, null, e, t);
}
function aa(e = "", t = !1) {
  return t ? (rr(), or(Ae, null, e)) : Xe(Ae, null, e);
}
function Ze(e, t) {
  let r = 0;
  const { shapeFlag: a } = e;
  if (t == null)
    t = null;
  else if (b(t))
    r = 16;
  else if (typeof t == "object")
    if (a & 65) {
      const o = t.default;
      o && (o._c && (o._d = !1), Ze(e, o()), o._c && (o._d = !0));
      return;
    } else {
      r = 32;
      const o = t._;
      !o && !(ar in t) ? t._ctx = T : o === 3 && T && (T.slots._ === 1 ? t._ = 1 : (t._ = 2, e.patchFlag |= 1024));
    }
  else
    v(t) ? (t = { default: t, _ctx: T }, r = 32) : (t = String(t), a & 64 ? (r = 16, t = [dr(t)]) : r = 8);
  e.children = t, e.shapeFlag |= r;
}
function na(...e) {
  const t = {};
  for (let r = 0; r < e.length; r++) {
    const a = e[r];
    for (const o in a)
      if (o === "class")
        t.class !== a.class && (t.class = Ue([t.class, a.class]));
      else if (o === "style")
        t.style = He([t.style, a.style]);
      else if (Or(o)) {
        const n = t[o], l = a[o];
        l && n !== l && !(b(n) && n.includes(l)) && (t[o] = n ? [].concat(n, l) : l);
      } else
        o !== "" && (t[o] = a[o]);
  }
  return t;
}
let G = null, et, L, xt = "__VUE_INSTANCE_SETTERS__";
(L = Ce()[xt]) || (L = Ce()[xt] = []), L.push((e) => G = e), et = (e) => {
  L.length > 1 ? L.forEach((t) => t(e)) : L[0](e);
};
const _t = (e) => {
  et(e), e.scope.on();
}, la = () => {
  G && G.scope.off(), et(null);
};
function ca(e) {
  return e.vnode.shapeFlag & 4;
}
function da(e) {
  if (e.exposed)
    return e.exposeProxy || (e.exposeProxy = new Proxy(go(bo(e.exposed)), {
      get(t, r) {
        if (r in t)
          return t[r];
        if (r in re)
          return re[r](e);
      },
      has(t, r) {
        return r in t || r in re;
      }
    }));
}
const ua = /(?:^|[-_])(\w)/g, ia = (e) => e.replace(ua, (t) => t.toUpperCase()).replace(/[-_]/g, "");
function tt(e, t = !0) {
  return v(e) ? e.displayName || e.name : e.name || t && e.__name;
}
function ur(e, t, r = !1) {
  let a = tt(t);
  if (!a && t.__file) {
    const o = t.__file.match(/([^/\\]+)\.\w+$/);
    o && (a = o[1]);
  }
  if (!a && e && e.parent) {
    const o = (n) => {
      for (const l in n)
        if (n[l] === t)
          return l;
    };
    a = o(
      e.components || e.parent.type.components
    ) || o(e.appContext.components);
  }
  return a ? ia(a) : r ? "App" : "Anonymous";
}
function ir(e) {
  return v(e) && "__vccOpts" in e;
}
function Se(e) {
  return !!(e && e.__v_isShallow);
}
function sa() {
  if (process.env.NODE_ENV === "production" || typeof window > "u")
    return;
  const e = { style: "color:#3ba776" }, t = { style: "color:#0b1bc9" }, r = { style: "color:#b62e24" }, a = { style: "color:#9d288c" }, o = {
    header(d) {
      return x(d) ? d.__isVue ? ["div", e, "VueInstance"] : E(d) ? [
        "div",
        {},
        ["span", e, h(d)],
        "<",
        c(d.value),
        ">"
      ] : W(d) ? [
        "div",
        {},
        ["span", e, Se(d) ? "ShallowReactive" : "Reactive"],
        "<",
        c(d),
        `>${q(d) ? " (readonly)" : ""}`
      ] : q(d) ? [
        "div",
        {},
        ["span", e, Se(d) ? "ShallowReadonly" : "Readonly"],
        "<",
        c(d),
        ">"
      ] : null : null;
    },
    hasBody(d) {
      return d && d.__isVue;
    },
    body(d) {
      if (d && d.__isVue)
        return [
          "div",
          {},
          ...n(d.$)
        ];
    }
  };
  function n(d) {
    const s = [];
    d.type.props && d.props && s.push(l("props", f(d.props))), d.setupState !== $ && s.push(l("setup", d.setupState)), d.data !== $ && s.push(l("data", f(d.data)));
    const m = u(d, "computed");
    m && s.push(l("computed", m));
    const y = u(d, "inject");
    return y && s.push(l("injected", y)), s.push([
      "div",
      {},
      [
        "span",
        {
          style: a.style + ";opacity:0.66"
        },
        "$ (internal): "
      ],
      ["object", { object: d }]
    ]), s;
  }
  function l(d, s) {
    return s = C({}, s), Object.keys(s).length ? [
      "div",
      { style: "line-height:1.25em;margin-bottom:0.6em" },
      [
        "div",
        {
          style: "color:#476582"
        },
        d
      ],
      [
        "div",
        {
          style: "padding-left:1.25em"
        },
        ...Object.keys(s).map((m) => [
          "div",
          {},
          ["span", a, m + ": "],
          c(s[m], !1)
        ])
      ]
    ] : ["span", {}];
  }
  function c(d, s = !0) {
    return typeof d == "number" ? ["span", t, d] : typeof d == "string" ? ["span", r, JSON.stringify(d)] : typeof d == "boolean" ? ["span", a, d] : x(d) ? ["object", { object: s ? f(d) : d }] : ["span", r, String(d)];
  }
  function u(d, s) {
    const m = d.type;
    if (v(m))
      return;
    const y = {};
    for (const O in d.ctx)
      i(m, O, s) && (y[O] = d.ctx[O]);
    return y;
  }
  function i(d, s, m) {
    const y = d[m];
    if (b(y) && y.includes(s) || x(y) && s in y || d.extends && i(d.extends, s, m) || d.mixins && d.mixins.some((O) => i(O, s, m)))
      return !0;
  }
  function h(d) {
    return Se(d) ? "ShallowRef" : d.effect ? "ComputedRef" : "Ref";
  }
  window.devtoolsFormatters ? window.devtoolsFormatters.push(o) : window.devtoolsFormatters = [o];
}
function fa() {
  sa();
}
process.env.NODE_ENV !== "production" && fa();
const ma = /* @__PURE__ */ Ko({
  __name: "ThemeStyles",
  props: {
    id: {}
  },
  setup(e) {
    return (t, r) => t.id !== "none" ? (rr(), or(Ho("style"), { key: 0 }, {
      default: Po(() => [
        dr(Mr(Wt(ha)(t.id)), 1)
      ]),
      _: 1
    })) : aa("", !0);
  }
}), sr = {
  alternate: br,
  default: wt,
  moon: vr,
  purple: xr,
  solarized: wr,
  bluePlanet: mr,
  deepSpace: gr,
  saturn: _r,
  kepler: pr,
  mars: kr
}, ga = Object.keys(sr), ha = (e = "default") => e === "none" ? "" : sr[e] ?? wt;
export {
  ma as ThemeStyles,
  ga as availableThemes,
  ha as getThemeById,
  sr as presets
};
